{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a4dd9be5-f6e1-42ad-8c4f-6acdf5e854d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(biomaRt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e248aea8-d68f-4147-8c5a-f87ea80792e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `gene` -> `gene...1`\n",
      "\u001b[36m•\u001b[39m `gene` -> `gene...11`\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m9661\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m26\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m (11): gene...1, groups_expressed, groups_not_expressed, gene...11, exps_...\n",
      "\u001b[32mdbl\u001b[39m (15): mean_exp, min_exp, max_exp, max_2nd, n_exp, frac_exp, lim, n_over,...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\u001b[1m\u001b[22mNew names:\n",
      "\u001b[36m•\u001b[39m `gene` -> `gene...1`\n",
      "\u001b[36m•\u001b[39m `gene` -> `gene...11`\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m17330\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m26\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m (11): gene...1, groups_expressed, groups_not_expressed, gene...11, exps_...\n",
      "\u001b[32mdbl\u001b[39m (15): mean_exp, min_exp, max_exp, max_2nd, n_exp, frac_exp, lim, n_over,...\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "xp_classes = read_csv(\"/nfs/research/irene/ysong/RESULTS/GeneCellType/GeneSpectra/zebrafish_xenopus_embryo/run_classification/xenopus_gene_name_ct_final.csv\")\n",
    "dr_classes = read_csv(\"/nfs/research/irene/ysong/RESULTS/GeneCellType/GeneSpectra/zebrafish_xenopus_embryo/run_classification/zebrafish_gene_name_ct_final.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a5dcb59a-bf21-4132-951f-fd081901142b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17330</li><li>26</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17330\n",
       "\\item 26\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17330\n",
       "2. 26\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17330    26"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(dr_classes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "300d7b80-7d75-4fe2-877c-91f8444da7b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene...1</th><th scope=col>mean_exp</th><th scope=col>min_exp</th><th scope=col>max_exp</th><th scope=col>max_2nd</th><th scope=col>n_exp</th><th scope=col>frac_exp</th><th scope=col>groups_expressed</th><th scope=col>groups_not_expressed</th><th scope=col>lim</th><th scope=col>⋯</th><th scope=col>enrichment_group</th><th scope=col>n_enriched</th><th scope=col>max_2nd_or_lim</th><th scope=col>exps_enhanced</th><th scope=col>n_enhanced</th><th scope=col>enhanced_in</th><th scope=col>spec_category</th><th scope=col>dist_category</th><th scope=col>spec_score</th><th scope=col>enriched_groups</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a2ml  </td><td>0.0042325</td><td>0</td><td>0.0232</td><td>0.0148</td><td>0</td><td>0.0</td><td>NA       </td><td>Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud</td><td>0.005800</td><td>⋯</td><td>NA       </td><td>0</td><td>0.014800</td><td>[]                </td><td>0</td><td>NA       </td><td>lowly expressed   </td><td>lowly expressed    </td><td>0.000000</td><td>NA       </td></tr>\n",
       "\t<tr><td>aars2 </td><td>0.0452075</td><td>0</td><td>0.1204</td><td>0.1070</td><td>0</td><td>0.0</td><td>NA       </td><td>Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud</td><td>0.030100</td><td>⋯</td><td>NA       </td><td>0</td><td>0.107000</td><td>[]                </td><td>0</td><td>NA       </td><td>lowly expressed   </td><td>lowly expressed    </td><td>0.000000</td><td>NA       </td></tr>\n",
       "\t<tr><td>abcb9 </td><td>0.0316125</td><td>0</td><td>0.5596</td><td>0.1588</td><td>0</td><td>0.0</td><td>NA       </td><td>Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud</td><td>0.139900</td><td>⋯</td><td>NA       </td><td>0</td><td>0.158800</td><td>[]                </td><td>0</td><td>NA       </td><td>lowly expressed   </td><td>lowly expressed    </td><td>0.000000</td><td>NA       </td></tr>\n",
       "\t<tr><td>abcc2 </td><td>0.0405875</td><td>0</td><td>0.6533</td><td>0.1370</td><td>0</td><td>0.0</td><td>NA       </td><td>Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud</td><td>0.163325</td><td>⋯</td><td>NA       </td><td>0</td><td>0.163325</td><td>[]                </td><td>0</td><td>NA       </td><td>lowly expressed   </td><td>lowly expressed    </td><td>0.000000</td><td>NA       </td></tr>\n",
       "\t<tr><td>abcg2c</td><td>0.3169100</td><td>0</td><td>1.3065</td><td>0.5839</td><td>1</td><td>2.5</td><td>Epiphysis</td><td>Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud          </td><td>0.326625</td><td>⋯</td><td>Epiphysis</td><td>1</td><td>0.583900</td><td>[1.30649995803833]</td><td>1</td><td>Epiphysis</td><td>cell type enhanced</td><td>expressed in single</td><td>4.122622</td><td>Epiphysis</td></tr>\n",
       "\t<tr><td>abcg8 </td><td>0.0449650</td><td>0</td><td>0.6533</td><td>0.1370</td><td>0</td><td>0.0</td><td>NA       </td><td>Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud</td><td>0.163325</td><td>⋯</td><td>NA       </td><td>0</td><td>0.163325</td><td>[]                </td><td>0</td><td>NA       </td><td>lowly expressed   </td><td>lowly expressed    </td><td>0.000000</td><td>NA       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 26\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene...1 & mean\\_exp & min\\_exp & max\\_exp & max\\_2nd & n\\_exp & frac\\_exp & groups\\_expressed & groups\\_not\\_expressed & lim & ⋯ & enrichment\\_group & n\\_enriched & max\\_2nd\\_or\\_lim & exps\\_enhanced & n\\_enhanced & enhanced\\_in & spec\\_category & dist\\_category & spec\\_score & enriched\\_groups\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <dbl> & ⋯ & <chr> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t a2ml   & 0.0042325 & 0 & 0.0232 & 0.0148 & 0 & 0.0 & NA        & Apoptotic\\_like;Blastula;Blood;Dorsal\\_organizer;Endoderm;Endothelial;Epidermal\\_progenitor;Epiphysis;Eye\\_primordium;Forebrain\\_midbrain;Forerunner\\_cells;Germline;Hatching\\_gland;Heart;Hindbrain;Intermediate\\_mesoderm;Involuting\\_marginal\\_zone;Ionocyte;Lens;Macrophage;Nanog\\_high;Neural\\_crest;Neuroectoderm;Neuron;Non\\_neural\\_ectoderm;Notochord;Notoplate;Olfactory\\_placode;Optic;Otic\\_placode;Pancreas\\_primordium;Pharyngeal\\_pouch;Placodal\\_area;Presomitic\\_mesoderm;Pronephric\\_duct;Rare\\_epidermal\\_subtypes;Rohon\\_beard\\_neuron;Secretory\\_epidermal;Skeletal\\_muscle;Tailbud & 0.005800 & ⋯ & NA        & 0 & 0.014800 & {[}{]}                 & 0 & NA        & lowly expressed    & lowly expressed     & 0.000000 & NA       \\\\\n",
       "\t aars2  & 0.0452075 & 0 & 0.1204 & 0.1070 & 0 & 0.0 & NA        & Apoptotic\\_like;Blastula;Blood;Dorsal\\_organizer;Endoderm;Endothelial;Epidermal\\_progenitor;Epiphysis;Eye\\_primordium;Forebrain\\_midbrain;Forerunner\\_cells;Germline;Hatching\\_gland;Heart;Hindbrain;Intermediate\\_mesoderm;Involuting\\_marginal\\_zone;Ionocyte;Lens;Macrophage;Nanog\\_high;Neural\\_crest;Neuroectoderm;Neuron;Non\\_neural\\_ectoderm;Notochord;Notoplate;Olfactory\\_placode;Optic;Otic\\_placode;Pancreas\\_primordium;Pharyngeal\\_pouch;Placodal\\_area;Presomitic\\_mesoderm;Pronephric\\_duct;Rare\\_epidermal\\_subtypes;Rohon\\_beard\\_neuron;Secretory\\_epidermal;Skeletal\\_muscle;Tailbud & 0.030100 & ⋯ & NA        & 0 & 0.107000 & {[}{]}                 & 0 & NA        & lowly expressed    & lowly expressed     & 0.000000 & NA       \\\\\n",
       "\t abcb9  & 0.0316125 & 0 & 0.5596 & 0.1588 & 0 & 0.0 & NA        & Apoptotic\\_like;Blastula;Blood;Dorsal\\_organizer;Endoderm;Endothelial;Epidermal\\_progenitor;Epiphysis;Eye\\_primordium;Forebrain\\_midbrain;Forerunner\\_cells;Germline;Hatching\\_gland;Heart;Hindbrain;Intermediate\\_mesoderm;Involuting\\_marginal\\_zone;Ionocyte;Lens;Macrophage;Nanog\\_high;Neural\\_crest;Neuroectoderm;Neuron;Non\\_neural\\_ectoderm;Notochord;Notoplate;Olfactory\\_placode;Optic;Otic\\_placode;Pancreas\\_primordium;Pharyngeal\\_pouch;Placodal\\_area;Presomitic\\_mesoderm;Pronephric\\_duct;Rare\\_epidermal\\_subtypes;Rohon\\_beard\\_neuron;Secretory\\_epidermal;Skeletal\\_muscle;Tailbud & 0.139900 & ⋯ & NA        & 0 & 0.158800 & {[}{]}                 & 0 & NA        & lowly expressed    & lowly expressed     & 0.000000 & NA       \\\\\n",
       "\t abcc2  & 0.0405875 & 0 & 0.6533 & 0.1370 & 0 & 0.0 & NA        & Apoptotic\\_like;Blastula;Blood;Dorsal\\_organizer;Endoderm;Endothelial;Epidermal\\_progenitor;Epiphysis;Eye\\_primordium;Forebrain\\_midbrain;Forerunner\\_cells;Germline;Hatching\\_gland;Heart;Hindbrain;Intermediate\\_mesoderm;Involuting\\_marginal\\_zone;Ionocyte;Lens;Macrophage;Nanog\\_high;Neural\\_crest;Neuroectoderm;Neuron;Non\\_neural\\_ectoderm;Notochord;Notoplate;Olfactory\\_placode;Optic;Otic\\_placode;Pancreas\\_primordium;Pharyngeal\\_pouch;Placodal\\_area;Presomitic\\_mesoderm;Pronephric\\_duct;Rare\\_epidermal\\_subtypes;Rohon\\_beard\\_neuron;Secretory\\_epidermal;Skeletal\\_muscle;Tailbud & 0.163325 & ⋯ & NA        & 0 & 0.163325 & {[}{]}                 & 0 & NA        & lowly expressed    & lowly expressed     & 0.000000 & NA       \\\\\n",
       "\t abcg2c & 0.3169100 & 0 & 1.3065 & 0.5839 & 1 & 2.5 & Epiphysis & Apoptotic\\_like;Blastula;Blood;Dorsal\\_organizer;Endoderm;Endothelial;Epidermal\\_progenitor;Eye\\_primordium;Forebrain\\_midbrain;Forerunner\\_cells;Germline;Hatching\\_gland;Heart;Hindbrain;Intermediate\\_mesoderm;Involuting\\_marginal\\_zone;Ionocyte;Lens;Macrophage;Nanog\\_high;Neural\\_crest;Neuroectoderm;Neuron;Non\\_neural\\_ectoderm;Notochord;Notoplate;Olfactory\\_placode;Optic;Otic\\_placode;Pancreas\\_primordium;Pharyngeal\\_pouch;Placodal\\_area;Presomitic\\_mesoderm;Pronephric\\_duct;Rare\\_epidermal\\_subtypes;Rohon\\_beard\\_neuron;Secretory\\_epidermal;Skeletal\\_muscle;Tailbud           & 0.326625 & ⋯ & Epiphysis & 1 & 0.583900 & {[}1.30649995803833{]} & 1 & Epiphysis & cell type enhanced & expressed in single & 4.122622 & Epiphysis\\\\\n",
       "\t abcg8  & 0.0449650 & 0 & 0.6533 & 0.1370 & 0 & 0.0 & NA        & Apoptotic\\_like;Blastula;Blood;Dorsal\\_organizer;Endoderm;Endothelial;Epidermal\\_progenitor;Epiphysis;Eye\\_primordium;Forebrain\\_midbrain;Forerunner\\_cells;Germline;Hatching\\_gland;Heart;Hindbrain;Intermediate\\_mesoderm;Involuting\\_marginal\\_zone;Ionocyte;Lens;Macrophage;Nanog\\_high;Neural\\_crest;Neuroectoderm;Neuron;Non\\_neural\\_ectoderm;Notochord;Notoplate;Olfactory\\_placode;Optic;Otic\\_placode;Pancreas\\_primordium;Pharyngeal\\_pouch;Placodal\\_area;Presomitic\\_mesoderm;Pronephric\\_duct;Rare\\_epidermal\\_subtypes;Rohon\\_beard\\_neuron;Secretory\\_epidermal;Skeletal\\_muscle;Tailbud & 0.163325 & ⋯ & NA        & 0 & 0.163325 & {[}{]}                 & 0 & NA        & lowly expressed    & lowly expressed     & 0.000000 & NA       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 26\n",
       "\n",
       "| gene...1 &lt;chr&gt; | mean_exp &lt;dbl&gt; | min_exp &lt;dbl&gt; | max_exp &lt;dbl&gt; | max_2nd &lt;dbl&gt; | n_exp &lt;dbl&gt; | frac_exp &lt;dbl&gt; | groups_expressed &lt;chr&gt; | groups_not_expressed &lt;chr&gt; | lim &lt;dbl&gt; | ⋯ ⋯ | enrichment_group &lt;chr&gt; | n_enriched &lt;dbl&gt; | max_2nd_or_lim &lt;dbl&gt; | exps_enhanced &lt;chr&gt; | n_enhanced &lt;dbl&gt; | enhanced_in &lt;chr&gt; | spec_category &lt;chr&gt; | dist_category &lt;chr&gt; | spec_score &lt;dbl&gt; | enriched_groups &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| a2ml   | 0.0042325 | 0 | 0.0232 | 0.0148 | 0 | 0.0 | NA        | Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud | 0.005800 | ⋯ | NA        | 0 | 0.014800 | []                 | 0 | NA        | lowly expressed    | lowly expressed     | 0.000000 | NA        |\n",
       "| aars2  | 0.0452075 | 0 | 0.1204 | 0.1070 | 0 | 0.0 | NA        | Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud | 0.030100 | ⋯ | NA        | 0 | 0.107000 | []                 | 0 | NA        | lowly expressed    | lowly expressed     | 0.000000 | NA        |\n",
       "| abcb9  | 0.0316125 | 0 | 0.5596 | 0.1588 | 0 | 0.0 | NA        | Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud | 0.139900 | ⋯ | NA        | 0 | 0.158800 | []                 | 0 | NA        | lowly expressed    | lowly expressed     | 0.000000 | NA        |\n",
       "| abcc2  | 0.0405875 | 0 | 0.6533 | 0.1370 | 0 | 0.0 | NA        | Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud | 0.163325 | ⋯ | NA        | 0 | 0.163325 | []                 | 0 | NA        | lowly expressed    | lowly expressed     | 0.000000 | NA        |\n",
       "| abcg2c | 0.3169100 | 0 | 1.3065 | 0.5839 | 1 | 2.5 | Epiphysis | Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud           | 0.326625 | ⋯ | Epiphysis | 1 | 0.583900 | [1.30649995803833] | 1 | Epiphysis | cell type enhanced | expressed in single | 4.122622 | Epiphysis |\n",
       "| abcg8  | 0.0449650 | 0 | 0.6533 | 0.1370 | 0 | 0.0 | NA        | Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud | 0.163325 | ⋯ | NA        | 0 | 0.163325 | []                 | 0 | NA        | lowly expressed    | lowly expressed     | 0.000000 | NA        |\n",
       "\n"
      ],
      "text/plain": [
       "  gene...1 mean_exp  min_exp max_exp max_2nd n_exp frac_exp groups_expressed\n",
       "1 a2ml     0.0042325 0       0.0232  0.0148  0     0.0      NA              \n",
       "2 aars2    0.0452075 0       0.1204  0.1070  0     0.0      NA              \n",
       "3 abcb9    0.0316125 0       0.5596  0.1588  0     0.0      NA              \n",
       "4 abcc2    0.0405875 0       0.6533  0.1370  0     0.0      NA              \n",
       "5 abcg2c   0.3169100 0       1.3065  0.5839  1     2.5      Epiphysis       \n",
       "6 abcg8    0.0449650 0       0.6533  0.1370  0     0.0      NA              \n",
       "  groups_not_expressed                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "1 Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud\n",
       "2 Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud\n",
       "3 Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud\n",
       "4 Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud\n",
       "5 Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud          \n",
       "6 Apoptotic_like;Blastula;Blood;Dorsal_organizer;Endoderm;Endothelial;Epidermal_progenitor;Epiphysis;Eye_primordium;Forebrain_midbrain;Forerunner_cells;Germline;Hatching_gland;Heart;Hindbrain;Intermediate_mesoderm;Involuting_marginal_zone;Ionocyte;Lens;Macrophage;Nanog_high;Neural_crest;Neuroectoderm;Neuron;Non_neural_ectoderm;Notochord;Notoplate;Olfactory_placode;Optic;Otic_placode;Pancreas_primordium;Pharyngeal_pouch;Placodal_area;Presomitic_mesoderm;Pronephric_duct;Rare_epidermal_subtypes;Rohon_beard_neuron;Secretory_epidermal;Skeletal_muscle;Tailbud\n",
       "  lim      ⋯ enrichment_group n_enriched max_2nd_or_lim exps_enhanced     \n",
       "1 0.005800 ⋯ NA               0          0.014800       []                \n",
       "2 0.030100 ⋯ NA               0          0.107000       []                \n",
       "3 0.139900 ⋯ NA               0          0.158800       []                \n",
       "4 0.163325 ⋯ NA               0          0.163325       []                \n",
       "5 0.326625 ⋯ Epiphysis        1          0.583900       [1.30649995803833]\n",
       "6 0.163325 ⋯ NA               0          0.163325       []                \n",
       "  n_enhanced enhanced_in spec_category      dist_category       spec_score\n",
       "1 0          NA          lowly expressed    lowly expressed     0.000000  \n",
       "2 0          NA          lowly expressed    lowly expressed     0.000000  \n",
       "3 0          NA          lowly expressed    lowly expressed     0.000000  \n",
       "4 0          NA          lowly expressed    lowly expressed     0.000000  \n",
       "5 1          Epiphysis   cell type enhanced expressed in single 4.122622  \n",
       "6 0          NA          lowly expressed    lowly expressed     0.000000  \n",
       "  enriched_groups\n",
       "1 NA             \n",
       "2 NA             \n",
       "3 NA             \n",
       "4 NA             \n",
       "5 Epiphysis      \n",
       "6 NA             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(dr_classes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f3308536-4715-4c4c-a636-421b36212b71",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>9661</li><li>26</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 9661\n",
       "\\item 26\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 9661\n",
       "2. 26\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 9661   26"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(xp_classes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b0db1af7-79f1-4931-9bd4-b76572d96e4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "species_1 = 'drerio'\n",
    "species_2 = 'xtropicalis'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "760a6e7a-e281-415a-8aa4-0c8423d268fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "## version = '105'\n",
    "mart <- useEnsembl(\"ensembl\", version = \"105\", dataset = as.character(paste(species_1, \"_gene_ensembl\", sep = \"\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "4e2a5852-bbe2-4328-a575-b988a0663e25",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# get genes in the main chrs of the first species\n",
    "genes_species_1 <- getBM(attributes = c(\"ensembl_gene_id\", \"external_gene_name\", \"chromosome_name\"), mart = mart) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ba2667c0-6dc5-4e5a-87ec-713d7adabf72",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3009 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>name</th><th scope=col>description</th><th scope=col>page</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ensembl_gene_id                </td><td>Gene stable ID                            </td><td>feature_page</td></tr>\n",
       "\t<tr><td>ensembl_gene_id_version        </td><td>Gene stable ID version                    </td><td>feature_page</td></tr>\n",
       "\t<tr><td>ensembl_transcript_id          </td><td>Transcript stable ID                      </td><td>feature_page</td></tr>\n",
       "\t<tr><td>ensembl_transcript_id_version  </td><td>Transcript stable ID version              </td><td>feature_page</td></tr>\n",
       "\t<tr><td>ensembl_peptide_id             </td><td>Protein stable ID                         </td><td>feature_page</td></tr>\n",
       "\t<tr><td>ensembl_peptide_id_version     </td><td>Protein stable ID version                 </td><td>feature_page</td></tr>\n",
       "\t<tr><td>ensembl_exon_id                </td><td>Exon stable ID                            </td><td>feature_page</td></tr>\n",
       "\t<tr><td>description                    </td><td>Gene description                          </td><td>feature_page</td></tr>\n",
       "\t<tr><td>chromosome_name                </td><td>Chromosome/scaffold name                  </td><td>feature_page</td></tr>\n",
       "\t<tr><td>start_position                 </td><td>Gene start (bp)                           </td><td>feature_page</td></tr>\n",
       "\t<tr><td>end_position                   </td><td>Gene end (bp)                             </td><td>feature_page</td></tr>\n",
       "\t<tr><td>strand                         </td><td>Strand                                    </td><td>feature_page</td></tr>\n",
       "\t<tr><td>band                           </td><td>Karyotype band                            </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_start               </td><td>Transcript start (bp)                     </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_end                 </td><td>Transcript end (bp)                       </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcription_start_site       </td><td>Transcription start site (TSS)            </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_length              </td><td>Transcript length (including UTRs and CDS)</td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_appris              </td><td>APPRIS annotation                         </td><td>feature_page</td></tr>\n",
       "\t<tr><td>external_gene_name             </td><td>Gene name                                 </td><td>feature_page</td></tr>\n",
       "\t<tr><td>external_gene_source           </td><td>Source of gene name                       </td><td>feature_page</td></tr>\n",
       "\t<tr><td>external_transcript_name       </td><td>Transcript name                           </td><td>feature_page</td></tr>\n",
       "\t<tr><td>external_transcript_source_name</td><td>Source of transcript name                 </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_count               </td><td>Transcript count                          </td><td>feature_page</td></tr>\n",
       "\t<tr><td>percentage_gene_gc_content     </td><td>Gene % GC content                         </td><td>feature_page</td></tr>\n",
       "\t<tr><td>gene_biotype                   </td><td>Gene type                                 </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_biotype             </td><td>Transcript type                           </td><td>feature_page</td></tr>\n",
       "\t<tr><td>source                         </td><td>Source (gene)                             </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_source              </td><td>Source (transcript)                       </td><td>feature_page</td></tr>\n",
       "\t<tr><td>version                        </td><td>Version (gene)                            </td><td>feature_page</td></tr>\n",
       "\t<tr><td>transcript_version             </td><td>Version (transcript)                      </td><td>feature_page</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>5_utr_end                    </td><td>5' UTR end                                </td><td>sequences</td></tr>\n",
       "\t<tr><td>3_utr_start                  </td><td>3' UTR start                              </td><td>sequences</td></tr>\n",
       "\t<tr><td>3_utr_end                    </td><td>3' UTR end                                </td><td>sequences</td></tr>\n",
       "\t<tr><td>ensembl_transcript_id        </td><td>Transcript stable ID                      </td><td>sequences</td></tr>\n",
       "\t<tr><td>ensembl_transcript_id_version</td><td>Transcript stable ID version              </td><td>sequences</td></tr>\n",
       "\t<tr><td>ensembl_peptide_id           </td><td>Protein stable ID                         </td><td>sequences</td></tr>\n",
       "\t<tr><td>ensembl_peptide_id_version   </td><td>Protein stable ID version                 </td><td>sequences</td></tr>\n",
       "\t<tr><td>transcript_biotype           </td><td>Transcript type                           </td><td>sequences</td></tr>\n",
       "\t<tr><td>transcript_version           </td><td>Version (transcript)                      </td><td>sequences</td></tr>\n",
       "\t<tr><td>peptide_version              </td><td>Version (protein)                         </td><td>sequences</td></tr>\n",
       "\t<tr><td>strand                       </td><td>Strand                                    </td><td>sequences</td></tr>\n",
       "\t<tr><td>transcript_start             </td><td>Transcript start (bp)                     </td><td>sequences</td></tr>\n",
       "\t<tr><td>transcript_end               </td><td>Transcript end (bp)                       </td><td>sequences</td></tr>\n",
       "\t<tr><td>transcription_start_site     </td><td>Transcription start site (TSS)            </td><td>sequences</td></tr>\n",
       "\t<tr><td>transcript_length            </td><td>Transcript length (including UTRs and CDS)</td><td>sequences</td></tr>\n",
       "\t<tr><td>cds_length                   </td><td>CDS Length                                </td><td>sequences</td></tr>\n",
       "\t<tr><td>cds_start                    </td><td>CDS start                                 </td><td>sequences</td></tr>\n",
       "\t<tr><td>cds_end                      </td><td>CDS end                                   </td><td>sequences</td></tr>\n",
       "\t<tr><td>ensembl_exon_id              </td><td>Exon stable ID                            </td><td>sequences</td></tr>\n",
       "\t<tr><td>exon_chrom_start             </td><td>Exon region start (bp)                    </td><td>sequences</td></tr>\n",
       "\t<tr><td>exon_chrom_end               </td><td>Exon region end (bp)                      </td><td>sequences</td></tr>\n",
       "\t<tr><td>strand                       </td><td>Strand                                    </td><td>sequences</td></tr>\n",
       "\t<tr><td>rank                         </td><td>Exon rank in transcript                   </td><td>sequences</td></tr>\n",
       "\t<tr><td>phase                        </td><td>Start phase                               </td><td>sequences</td></tr>\n",
       "\t<tr><td>end_phase                    </td><td>End phase                                 </td><td>sequences</td></tr>\n",
       "\t<tr><td>cdna_coding_start            </td><td>cDNA coding start                         </td><td>sequences</td></tr>\n",
       "\t<tr><td>cdna_coding_end              </td><td>cDNA coding end                           </td><td>sequences</td></tr>\n",
       "\t<tr><td>genomic_coding_start         </td><td>Genomic coding start                      </td><td>sequences</td></tr>\n",
       "\t<tr><td>genomic_coding_end           </td><td>Genomic coding end                        </td><td>sequences</td></tr>\n",
       "\t<tr><td>is_constitutive              </td><td>Constitutive exon                         </td><td>sequences</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3009 × 3\n",
       "\\begin{tabular}{lll}\n",
       " name & description & page\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ensembl\\_gene\\_id                 & Gene stable ID                             & feature\\_page\\\\\n",
       "\t ensembl\\_gene\\_id\\_version         & Gene stable ID version                     & feature\\_page\\\\\n",
       "\t ensembl\\_transcript\\_id           & Transcript stable ID                       & feature\\_page\\\\\n",
       "\t ensembl\\_transcript\\_id\\_version   & Transcript stable ID version               & feature\\_page\\\\\n",
       "\t ensembl\\_peptide\\_id              & Protein stable ID                          & feature\\_page\\\\\n",
       "\t ensembl\\_peptide\\_id\\_version      & Protein stable ID version                  & feature\\_page\\\\\n",
       "\t ensembl\\_exon\\_id                 & Exon stable ID                             & feature\\_page\\\\\n",
       "\t description                     & Gene description                           & feature\\_page\\\\\n",
       "\t chromosome\\_name                 & Chromosome/scaffold name                   & feature\\_page\\\\\n",
       "\t start\\_position                  & Gene start (bp)                            & feature\\_page\\\\\n",
       "\t end\\_position                    & Gene end (bp)                              & feature\\_page\\\\\n",
       "\t strand                          & Strand                                     & feature\\_page\\\\\n",
       "\t band                            & Karyotype band                             & feature\\_page\\\\\n",
       "\t transcript\\_start                & Transcript start (bp)                      & feature\\_page\\\\\n",
       "\t transcript\\_end                  & Transcript end (bp)                        & feature\\_page\\\\\n",
       "\t transcription\\_start\\_site        & Transcription start site (TSS)             & feature\\_page\\\\\n",
       "\t transcript\\_length               & Transcript length (including UTRs and CDS) & feature\\_page\\\\\n",
       "\t transcript\\_appris               & APPRIS annotation                          & feature\\_page\\\\\n",
       "\t external\\_gene\\_name              & Gene name                                  & feature\\_page\\\\\n",
       "\t external\\_gene\\_source            & Source of gene name                        & feature\\_page\\\\\n",
       "\t external\\_transcript\\_name        & Transcript name                            & feature\\_page\\\\\n",
       "\t external\\_transcript\\_source\\_name & Source of transcript name                  & feature\\_page\\\\\n",
       "\t transcript\\_count                & Transcript count                           & feature\\_page\\\\\n",
       "\t percentage\\_gene\\_gc\\_content      & Gene \\% GC content                          & feature\\_page\\\\\n",
       "\t gene\\_biotype                    & Gene type                                  & feature\\_page\\\\\n",
       "\t transcript\\_biotype              & Transcript type                            & feature\\_page\\\\\n",
       "\t source                          & Source (gene)                              & feature\\_page\\\\\n",
       "\t transcript\\_source               & Source (transcript)                        & feature\\_page\\\\\n",
       "\t version                         & Version (gene)                             & feature\\_page\\\\\n",
       "\t transcript\\_version              & Version (transcript)                       & feature\\_page\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t 5\\_utr\\_end                     & 5' UTR end                                 & sequences\\\\\n",
       "\t 3\\_utr\\_start                   & 3' UTR start                               & sequences\\\\\n",
       "\t 3\\_utr\\_end                     & 3' UTR end                                 & sequences\\\\\n",
       "\t ensembl\\_transcript\\_id         & Transcript stable ID                       & sequences\\\\\n",
       "\t ensembl\\_transcript\\_id\\_version & Transcript stable ID version               & sequences\\\\\n",
       "\t ensembl\\_peptide\\_id            & Protein stable ID                          & sequences\\\\\n",
       "\t ensembl\\_peptide\\_id\\_version    & Protein stable ID version                  & sequences\\\\\n",
       "\t transcript\\_biotype            & Transcript type                            & sequences\\\\\n",
       "\t transcript\\_version            & Version (transcript)                       & sequences\\\\\n",
       "\t peptide\\_version               & Version (protein)                          & sequences\\\\\n",
       "\t strand                        & Strand                                     & sequences\\\\\n",
       "\t transcript\\_start              & Transcript start (bp)                      & sequences\\\\\n",
       "\t transcript\\_end                & Transcript end (bp)                        & sequences\\\\\n",
       "\t transcription\\_start\\_site      & Transcription start site (TSS)             & sequences\\\\\n",
       "\t transcript\\_length             & Transcript length (including UTRs and CDS) & sequences\\\\\n",
       "\t cds\\_length                    & CDS Length                                 & sequences\\\\\n",
       "\t cds\\_start                     & CDS start                                  & sequences\\\\\n",
       "\t cds\\_end                       & CDS end                                    & sequences\\\\\n",
       "\t ensembl\\_exon\\_id               & Exon stable ID                             & sequences\\\\\n",
       "\t exon\\_chrom\\_start              & Exon region start (bp)                     & sequences\\\\\n",
       "\t exon\\_chrom\\_end                & Exon region end (bp)                       & sequences\\\\\n",
       "\t strand                        & Strand                                     & sequences\\\\\n",
       "\t rank                          & Exon rank in transcript                    & sequences\\\\\n",
       "\t phase                         & Start phase                                & sequences\\\\\n",
       "\t end\\_phase                     & End phase                                  & sequences\\\\\n",
       "\t cdna\\_coding\\_start             & cDNA coding start                          & sequences\\\\\n",
       "\t cdna\\_coding\\_end               & cDNA coding end                            & sequences\\\\\n",
       "\t genomic\\_coding\\_start          & Genomic coding start                       & sequences\\\\\n",
       "\t genomic\\_coding\\_end            & Genomic coding end                         & sequences\\\\\n",
       "\t is\\_constitutive               & Constitutive exon                          & sequences\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3009 × 3\n",
       "\n",
       "| name &lt;chr&gt; | description &lt;chr&gt; | page &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ensembl_gene_id                 | Gene stable ID                             | feature_page |\n",
       "| ensembl_gene_id_version         | Gene stable ID version                     | feature_page |\n",
       "| ensembl_transcript_id           | Transcript stable ID                       | feature_page |\n",
       "| ensembl_transcript_id_version   | Transcript stable ID version               | feature_page |\n",
       "| ensembl_peptide_id              | Protein stable ID                          | feature_page |\n",
       "| ensembl_peptide_id_version      | Protein stable ID version                  | feature_page |\n",
       "| ensembl_exon_id                 | Exon stable ID                             | feature_page |\n",
       "| description                     | Gene description                           | feature_page |\n",
       "| chromosome_name                 | Chromosome/scaffold name                   | feature_page |\n",
       "| start_position                  | Gene start (bp)                            | feature_page |\n",
       "| end_position                    | Gene end (bp)                              | feature_page |\n",
       "| strand                          | Strand                                     | feature_page |\n",
       "| band                            | Karyotype band                             | feature_page |\n",
       "| transcript_start                | Transcript start (bp)                      | feature_page |\n",
       "| transcript_end                  | Transcript end (bp)                        | feature_page |\n",
       "| transcription_start_site        | Transcription start site (TSS)             | feature_page |\n",
       "| transcript_length               | Transcript length (including UTRs and CDS) | feature_page |\n",
       "| transcript_appris               | APPRIS annotation                          | feature_page |\n",
       "| external_gene_name              | Gene name                                  | feature_page |\n",
       "| external_gene_source            | Source of gene name                        | feature_page |\n",
       "| external_transcript_name        | Transcript name                            | feature_page |\n",
       "| external_transcript_source_name | Source of transcript name                  | feature_page |\n",
       "| transcript_count                | Transcript count                           | feature_page |\n",
       "| percentage_gene_gc_content      | Gene % GC content                          | feature_page |\n",
       "| gene_biotype                    | Gene type                                  | feature_page |\n",
       "| transcript_biotype              | Transcript type                            | feature_page |\n",
       "| source                          | Source (gene)                              | feature_page |\n",
       "| transcript_source               | Source (transcript)                        | feature_page |\n",
       "| version                         | Version (gene)                             | feature_page |\n",
       "| transcript_version              | Version (transcript)                       | feature_page |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| 5_utr_end                     | 5' UTR end                                 | sequences |\n",
       "| 3_utr_start                   | 3' UTR start                               | sequences |\n",
       "| 3_utr_end                     | 3' UTR end                                 | sequences |\n",
       "| ensembl_transcript_id         | Transcript stable ID                       | sequences |\n",
       "| ensembl_transcript_id_version | Transcript stable ID version               | sequences |\n",
       "| ensembl_peptide_id            | Protein stable ID                          | sequences |\n",
       "| ensembl_peptide_id_version    | Protein stable ID version                  | sequences |\n",
       "| transcript_biotype            | Transcript type                            | sequences |\n",
       "| transcript_version            | Version (transcript)                       | sequences |\n",
       "| peptide_version               | Version (protein)                          | sequences |\n",
       "| strand                        | Strand                                     | sequences |\n",
       "| transcript_start              | Transcript start (bp)                      | sequences |\n",
       "| transcript_end                | Transcript end (bp)                        | sequences |\n",
       "| transcription_start_site      | Transcription start site (TSS)             | sequences |\n",
       "| transcript_length             | Transcript length (including UTRs and CDS) | sequences |\n",
       "| cds_length                    | CDS Length                                 | sequences |\n",
       "| cds_start                     | CDS start                                  | sequences |\n",
       "| cds_end                       | CDS end                                    | sequences |\n",
       "| ensembl_exon_id               | Exon stable ID                             | sequences |\n",
       "| exon_chrom_start              | Exon region start (bp)                     | sequences |\n",
       "| exon_chrom_end                | Exon region end (bp)                       | sequences |\n",
       "| strand                        | Strand                                     | sequences |\n",
       "| rank                          | Exon rank in transcript                    | sequences |\n",
       "| phase                         | Start phase                                | sequences |\n",
       "| end_phase                     | End phase                                  | sequences |\n",
       "| cdna_coding_start             | cDNA coding start                          | sequences |\n",
       "| cdna_coding_end               | cDNA coding end                            | sequences |\n",
       "| genomic_coding_start          | Genomic coding start                       | sequences |\n",
       "| genomic_coding_end            | Genomic coding end                         | sequences |\n",
       "| is_constitutive               | Constitutive exon                          | sequences |\n",
       "\n"
      ],
      "text/plain": [
       "     name                            description                               \n",
       "1    ensembl_gene_id                 Gene stable ID                            \n",
       "2    ensembl_gene_id_version         Gene stable ID version                    \n",
       "3    ensembl_transcript_id           Transcript stable ID                      \n",
       "4    ensembl_transcript_id_version   Transcript stable ID version              \n",
       "5    ensembl_peptide_id              Protein stable ID                         \n",
       "6    ensembl_peptide_id_version      Protein stable ID version                 \n",
       "7    ensembl_exon_id                 Exon stable ID                            \n",
       "8    description                     Gene description                          \n",
       "9    chromosome_name                 Chromosome/scaffold name                  \n",
       "10   start_position                  Gene start (bp)                           \n",
       "11   end_position                    Gene end (bp)                             \n",
       "12   strand                          Strand                                    \n",
       "13   band                            Karyotype band                            \n",
       "14   transcript_start                Transcript start (bp)                     \n",
       "15   transcript_end                  Transcript end (bp)                       \n",
       "16   transcription_start_site        Transcription start site (TSS)            \n",
       "17   transcript_length               Transcript length (including UTRs and CDS)\n",
       "18   transcript_appris               APPRIS annotation                         \n",
       "19   external_gene_name              Gene name                                 \n",
       "20   external_gene_source            Source of gene name                       \n",
       "21   external_transcript_name        Transcript name                           \n",
       "22   external_transcript_source_name Source of transcript name                 \n",
       "23   transcript_count                Transcript count                          \n",
       "24   percentage_gene_gc_content      Gene % GC content                         \n",
       "25   gene_biotype                    Gene type                                 \n",
       "26   transcript_biotype              Transcript type                           \n",
       "27   source                          Source (gene)                             \n",
       "28   transcript_source               Source (transcript)                       \n",
       "29   version                         Version (gene)                            \n",
       "30   transcript_version              Version (transcript)                      \n",
       "⋮    ⋮                               ⋮                                         \n",
       "2980 5_utr_end                       5' UTR end                                \n",
       "2981 3_utr_start                     3' UTR start                              \n",
       "2982 3_utr_end                       3' UTR end                                \n",
       "2983 ensembl_transcript_id           Transcript stable ID                      \n",
       "2984 ensembl_transcript_id_version   Transcript stable ID version              \n",
       "2985 ensembl_peptide_id              Protein stable ID                         \n",
       "2986 ensembl_peptide_id_version      Protein stable ID version                 \n",
       "2987 transcript_biotype              Transcript type                           \n",
       "2988 transcript_version              Version (transcript)                      \n",
       "2989 peptide_version                 Version (protein)                         \n",
       "2990 strand                          Strand                                    \n",
       "2991 transcript_start                Transcript start (bp)                     \n",
       "2992 transcript_end                  Transcript end (bp)                       \n",
       "2993 transcription_start_site        Transcription start site (TSS)            \n",
       "2994 transcript_length               Transcript length (including UTRs and CDS)\n",
       "2995 cds_length                      CDS Length                                \n",
       "2996 cds_start                       CDS start                                 \n",
       "2997 cds_end                         CDS end                                   \n",
       "2998 ensembl_exon_id                 Exon stable ID                            \n",
       "2999 exon_chrom_start                Exon region start (bp)                    \n",
       "3000 exon_chrom_end                  Exon region end (bp)                      \n",
       "3001 strand                          Strand                                    \n",
       "3002 rank                            Exon rank in transcript                   \n",
       "3003 phase                           Start phase                               \n",
       "3004 end_phase                       End phase                                 \n",
       "3005 cdna_coding_start               cDNA coding start                         \n",
       "3006 cdna_coding_end                 cDNA coding end                           \n",
       "3007 genomic_coding_start            Genomic coding start                      \n",
       "3008 genomic_coding_end              Genomic coding end                        \n",
       "3009 is_constitutive                 Constitutive exon                         \n",
       "     page        \n",
       "1    feature_page\n",
       "2    feature_page\n",
       "3    feature_page\n",
       "4    feature_page\n",
       "5    feature_page\n",
       "6    feature_page\n",
       "7    feature_page\n",
       "8    feature_page\n",
       "9    feature_page\n",
       "10   feature_page\n",
       "11   feature_page\n",
       "12   feature_page\n",
       "13   feature_page\n",
       "14   feature_page\n",
       "15   feature_page\n",
       "16   feature_page\n",
       "17   feature_page\n",
       "18   feature_page\n",
       "19   feature_page\n",
       "20   feature_page\n",
       "21   feature_page\n",
       "22   feature_page\n",
       "23   feature_page\n",
       "24   feature_page\n",
       "25   feature_page\n",
       "26   feature_page\n",
       "27   feature_page\n",
       "28   feature_page\n",
       "29   feature_page\n",
       "30   feature_page\n",
       "⋮    ⋮           \n",
       "2980 sequences   \n",
       "2981 sequences   \n",
       "2982 sequences   \n",
       "2983 sequences   \n",
       "2984 sequences   \n",
       "2985 sequences   \n",
       "2986 sequences   \n",
       "2987 sequences   \n",
       "2988 sequences   \n",
       "2989 sequences   \n",
       "2990 sequences   \n",
       "2991 sequences   \n",
       "2992 sequences   \n",
       "2993 sequences   \n",
       "2994 sequences   \n",
       "2995 sequences   \n",
       "2996 sequences   \n",
       "2997 sequences   \n",
       "2998 sequences   \n",
       "2999 sequences   \n",
       "3000 sequences   \n",
       "3001 sequences   \n",
       "3002 sequences   \n",
       "3003 sequences   \n",
       "3004 sequences   \n",
       "3005 sequences   \n",
       "3006 sequences   \n",
       "3007 sequences   \n",
       "3008 sequences   \n",
       "3009 sequences   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "listAttributes(mart)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "735e1e52-c9d4-4eef-ad51-aabe2cf31483",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "avail_attributes <- listAttributes(mart) %>% filter(grepl(species_2, name)) %>% filter(grepl(\"homo\", name)) %>%\n",
    "filter(!grepl(\"Query protein or transcript ID\", description))\n",
    "biomartCacheClear()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "d9df7092-843a-4dcd-9f52-00be0e4bc983",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 13 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>name</th><th scope=col>description</th><th scope=col>page</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>xtropicalis_homolog_ensembl_gene        </td><td>Tropical clawed frog gene stable ID                          </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_associated_gene_name</td><td>Tropical clawed frog gene name                               </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_ensembl_peptide     </td><td>Tropical clawed frog protein or transcript stable ID         </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_chromosome          </td><td>Tropical clawed frog chromosome/scaffold name                </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_chrom_start         </td><td>Tropical clawed frog chromosome/scaffold start (bp)          </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_chrom_end           </td><td>Tropical clawed frog chromosome/scaffold end (bp)            </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_subtype             </td><td>Last common ancestor with Tropical clawed frog               </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_orthology_type      </td><td>Tropical clawed frog homology type                           </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_perc_id             </td><td>%id. target Tropical clawed frog gene identical to query gene</td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_perc_id_r1          </td><td>%id. query gene identical to target Tropical clawed frog gene</td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_goc_score           </td><td>Tropical clawed frog Gene-order conservation score           </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_wga_coverage        </td><td>Tropical clawed frog Whole-genome alignment coverage         </td><td>homologs</td></tr>\n",
       "\t<tr><td>xtropicalis_homolog_orthology_confidence</td><td>Tropical clawed frog orthology confidence [0 low, 1 high]    </td><td>homologs</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 13 × 3\n",
       "\\begin{tabular}{lll}\n",
       " name & description & page\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t xtropicalis\\_homolog\\_ensembl\\_gene         & Tropical clawed frog gene stable ID                           & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_associated\\_gene\\_name & Tropical clawed frog gene name                                & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_ensembl\\_peptide      & Tropical clawed frog protein or transcript stable ID          & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_chromosome           & Tropical clawed frog chromosome/scaffold name                 & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_chrom\\_start          & Tropical clawed frog chromosome/scaffold start (bp)           & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_chrom\\_end            & Tropical clawed frog chromosome/scaffold end (bp)             & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_subtype              & Last common ancestor with Tropical clawed frog                & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_orthology\\_type       & Tropical clawed frog homology type                            & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_perc\\_id              & \\%id. target Tropical clawed frog gene identical to query gene & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_perc\\_id\\_r1           & \\%id. query gene identical to target Tropical clawed frog gene & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_goc\\_score            & Tropical clawed frog Gene-order conservation score            & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_wga\\_coverage         & Tropical clawed frog Whole-genome alignment coverage          & homologs\\\\\n",
       "\t xtropicalis\\_homolog\\_orthology\\_confidence & Tropical clawed frog orthology confidence {[}0 low, 1 high{]}     & homologs\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 13 × 3\n",
       "\n",
       "| name &lt;chr&gt; | description &lt;chr&gt; | page &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| xtropicalis_homolog_ensembl_gene         | Tropical clawed frog gene stable ID                           | homologs |\n",
       "| xtropicalis_homolog_associated_gene_name | Tropical clawed frog gene name                                | homologs |\n",
       "| xtropicalis_homolog_ensembl_peptide      | Tropical clawed frog protein or transcript stable ID          | homologs |\n",
       "| xtropicalis_homolog_chromosome           | Tropical clawed frog chromosome/scaffold name                 | homologs |\n",
       "| xtropicalis_homolog_chrom_start          | Tropical clawed frog chromosome/scaffold start (bp)           | homologs |\n",
       "| xtropicalis_homolog_chrom_end            | Tropical clawed frog chromosome/scaffold end (bp)             | homologs |\n",
       "| xtropicalis_homolog_subtype              | Last common ancestor with Tropical clawed frog                | homologs |\n",
       "| xtropicalis_homolog_orthology_type       | Tropical clawed frog homology type                            | homologs |\n",
       "| xtropicalis_homolog_perc_id              | %id. target Tropical clawed frog gene identical to query gene | homologs |\n",
       "| xtropicalis_homolog_perc_id_r1           | %id. query gene identical to target Tropical clawed frog gene | homologs |\n",
       "| xtropicalis_homolog_goc_score            | Tropical clawed frog Gene-order conservation score            | homologs |\n",
       "| xtropicalis_homolog_wga_coverage         | Tropical clawed frog Whole-genome alignment coverage          | homologs |\n",
       "| xtropicalis_homolog_orthology_confidence | Tropical clawed frog orthology confidence [0 low, 1 high]     | homologs |\n",
       "\n"
      ],
      "text/plain": [
       "   name                                    \n",
       "1  xtropicalis_homolog_ensembl_gene        \n",
       "2  xtropicalis_homolog_associated_gene_name\n",
       "3  xtropicalis_homolog_ensembl_peptide     \n",
       "4  xtropicalis_homolog_chromosome          \n",
       "5  xtropicalis_homolog_chrom_start         \n",
       "6  xtropicalis_homolog_chrom_end           \n",
       "7  xtropicalis_homolog_subtype             \n",
       "8  xtropicalis_homolog_orthology_type      \n",
       "9  xtropicalis_homolog_perc_id             \n",
       "10 xtropicalis_homolog_perc_id_r1          \n",
       "11 xtropicalis_homolog_goc_score           \n",
       "12 xtropicalis_homolog_wga_coverage        \n",
       "13 xtropicalis_homolog_orthology_confidence\n",
       "   description                                                   page    \n",
       "1  Tropical clawed frog gene stable ID                           homologs\n",
       "2  Tropical clawed frog gene name                                homologs\n",
       "3  Tropical clawed frog protein or transcript stable ID          homologs\n",
       "4  Tropical clawed frog chromosome/scaffold name                 homologs\n",
       "5  Tropical clawed frog chromosome/scaffold start (bp)           homologs\n",
       "6  Tropical clawed frog chromosome/scaffold end (bp)             homologs\n",
       "7  Last common ancestor with Tropical clawed frog                homologs\n",
       "8  Tropical clawed frog homology type                            homologs\n",
       "9  %id. target Tropical clawed frog gene identical to query gene homologs\n",
       "10 %id. query gene identical to target Tropical clawed frog gene homologs\n",
       "11 Tropical clawed frog Gene-order conservation score            homologs\n",
       "12 Tropical clawed frog Whole-genome alignment coverage          homologs\n",
       "13 Tropical clawed frog orthology confidence [0 low, 1 high]     homologs"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "avail_attributes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "941f7a89-1d37-4662-ba56-0efca230f0a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "homology_tbl = getBM(attributes = c(\"ensembl_gene_id\", \"external_gene_name\", \"chromosome_name\", \"start_position\", \"end_position\", \"ensembl_peptide_id\", \n",
    "    avail_attributes$name), mart = mart, filters = \"external_gene_name\",\n",
    "  values = dr_classes[[\"gene...1\"]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "ee4ab12b-b9d3-48aa-ad40-a96b99870882",
   "metadata": {},
   "outputs": [],
   "source": [
    "homology_tbl %>% filter(external_gene_name %in% dr_classes[[\"gene...1\"]]) %>% filter(xtropicalis_homolog_associated_gene_name\t%in% xp_classes[['gene...1']]) %>% \n",
    "write_csv(\"drerio_xtropialis_ensembl_orthology_filtered.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "1938b813-9920-4901-8571-2a33b7ac5693",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 43372 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ensembl_gene_id</th><th scope=col>external_gene_name</th><th scope=col>chromosome_name</th><th scope=col>start_position</th><th scope=col>end_position</th><th scope=col>ensembl_peptide_id</th><th scope=col>xtropicalis_homolog_ensembl_gene</th><th scope=col>xtropicalis_homolog_associated_gene_name</th><th scope=col>xtropicalis_homolog_ensembl_peptide</th><th scope=col>xtropicalis_homolog_chromosome</th><th scope=col>xtropicalis_homolog_chrom_start</th><th scope=col>xtropicalis_homolog_chrom_end</th><th scope=col>xtropicalis_homolog_subtype</th><th scope=col>xtropicalis_homolog_orthology_type</th><th scope=col>xtropicalis_homolog_perc_id</th><th scope=col>xtropicalis_homolog_perc_id_r1</th><th scope=col>xtropicalis_homolog_goc_score</th><th scope=col>xtropicalis_homolog_wga_coverage</th><th scope=col>xtropicalis_homolog_orthology_confidence</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSDARG00000063344</td><td>fam162a </td><td>24</td><td>20927135</td><td>20934666</td><td>ENSDARP00000123357</td><td>ENSXETG00000026898</td><td>fam162a </td><td>ENSXETP00000056418</td><td>KV460586.1</td><td>     691</td><td>    4121</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>35.8974</td><td>47.8632</td><td>25</td><td> 68.86</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000063344</td><td>fam162a </td><td>24</td><td>20927135</td><td>20934666</td><td>ENSDARP00000114467</td><td>ENSXETG00000026898</td><td>fam162a </td><td>ENSXETP00000056418</td><td>KV460586.1</td><td>     691</td><td>    4121</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>35.8974</td><td>47.8632</td><td>25</td><td> 68.86</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000103391</td><td>rab9b   </td><td>14</td><td>14603731</td><td>14607855</td><td>ENSDARP00000140695</td><td>ENSXETG00000018133</td><td>rab9b   </td><td>ENSXETP00000039329</td><td>8         </td><td>16957503</td><td>16961130</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>75.2475</td><td>75.6219</td><td>25</td><td>100.00</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000103391</td><td>rab9b   </td><td>14</td><td>14603731</td><td>14607855</td><td>ENSDARP00000147450</td><td>ENSXETG00000018133</td><td>rab9b   </td><td>ENSXETP00000039329</td><td>8         </td><td>16957503</td><td>16961130</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>75.2475</td><td>75.6219</td><td>25</td><td>100.00</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000080018</td><td>kif16bb </td><td>13</td><td>34627550</td><td>34683370</td><td>ENSDARP00000099496</td><td>                  </td><td>        </td><td>                  </td><td>          </td><td>      NA</td><td>      NA</td><td>             </td><td>                  </td><td>     NA</td><td>     NA</td><td>NA</td><td>    NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000036109</td><td>mtx1b   </td><td>19</td><td>24448584</td><td>24484758</td><td>ENSDARP00000052421</td><td>                  </td><td>        </td><td>                  </td><td>          </td><td>      NA</td><td>      NA</td><td>             </td><td>                  </td><td>     NA</td><td>     NA</td><td>NA</td><td>    NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000075759</td><td>samd4a  </td><td>17</td><td>15033822</td><td>15117508</td><td>ENSDARP00000129251</td><td>ENSXETG00000012217</td><td>samd4a  </td><td>ENSXETP00000073845</td><td>8         </td><td>74362645</td><td>74457225</td><td>Gnathostomata</td><td>ortholog_one2one  </td><td>70.0581</td><td>67.6966</td><td>50</td><td> 80.53</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000075759</td><td>samd4a  </td><td>17</td><td>15033822</td><td>15117508</td><td>ENSDARP00000101105</td><td>ENSXETG00000012217</td><td>samd4a  </td><td>ENSXETP00000073845</td><td>8         </td><td>74362645</td><td>74457225</td><td>Gnathostomata</td><td>ortholog_one2one  </td><td>70.0581</td><td>67.6966</td><td>50</td><td> 80.53</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000015793</td><td>creb3l1 </td><td>7 </td><td>38897836</td><td>38954874</td><td>ENSDARP00000012411</td><td>ENSXETG00000031075</td><td>creb3l1 </td><td>ENSXETP00000062801</td><td>4         </td><td> 5396261</td><td> 5434390</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>51.9011</td><td>58.4582</td><td>75</td><td> 60.05</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000015793</td><td>creb3l1 </td><td>7 </td><td>38897836</td><td>38954874</td><td>ENSDARP00000142646</td><td>ENSXETG00000031075</td><td>creb3l1 </td><td>ENSXETP00000062801</td><td>4         </td><td> 5396261</td><td> 5434390</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>51.9011</td><td>58.4582</td><td>75</td><td> 60.05</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000015793</td><td>creb3l1 </td><td>7 </td><td>38897836</td><td>38954874</td><td>ENSDARP00000134652</td><td>ENSXETG00000031075</td><td>creb3l1 </td><td>ENSXETP00000062801</td><td>4         </td><td> 5396261</td><td> 5434390</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>51.9011</td><td>58.4582</td><td>75</td><td> 60.05</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000044573</td><td>cdc42   </td><td>11</td><td>39120306</td><td>39148199</td><td>ENSDARP00000129339</td><td>ENSXETG00000004743</td><td>cdc42   </td><td>ENSXETP00000030366</td><td>7         </td><td>68594070</td><td>68619099</td><td>Chordata     </td><td>ortholog_one2one  </td><td>98.4293</td><td>88.6792</td><td>75</td><td> 94.10</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000044573</td><td>cdc42   </td><td>11</td><td>39120306</td><td>39148199</td><td>ENSDARP00000129719</td><td>ENSXETG00000004743</td><td>cdc42   </td><td>ENSXETP00000030366</td><td>7         </td><td>68594070</td><td>68619099</td><td>Chordata     </td><td>ortholog_one2one  </td><td>98.4293</td><td>88.6792</td><td>75</td><td> 94.10</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000044573</td><td>cdc42   </td><td>11</td><td>39120306</td><td>39148199</td><td>ENSDARP00000149720</td><td>ENSXETG00000004743</td><td>cdc42   </td><td>ENSXETP00000030366</td><td>7         </td><td>68594070</td><td>68619099</td><td>Chordata     </td><td>ortholog_one2one  </td><td>98.4293</td><td>88.6792</td><td>75</td><td> 94.10</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000044573</td><td>cdc42   </td><td>11</td><td>39120306</td><td>39148199</td><td>ENSDARP00000155235</td><td>ENSXETG00000004743</td><td>cdc42   </td><td>ENSXETP00000030366</td><td>7         </td><td>68594070</td><td>68619099</td><td>Chordata     </td><td>ortholog_one2one  </td><td>98.4293</td><td>88.6792</td><td>75</td><td> 94.10</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000076270</td><td>adamts13</td><td>8 </td><td>45226586</td><td>45279411</td><td>ENSDARP00000119445</td><td>ENSXETG00000006882</td><td>adamts13</td><td>ENSXETP00000070357</td><td>KV460557.1</td><td>   60725</td><td>   78242</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>38.6515</td><td>42.8889</td><td> 0</td><td> 35.79</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000076270</td><td>adamts13</td><td>8 </td><td>45226586</td><td>45279411</td><td>ENSDARP00000143739</td><td>ENSXETG00000006882</td><td>adamts13</td><td>ENSXETP00000070357</td><td>KV460557.1</td><td>   60725</td><td>   78242</td><td>Euteleostomi </td><td>ortholog_one2one  </td><td>38.6515</td><td>42.8889</td><td> 0</td><td> 35.79</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000045415</td><td>gnal    </td><td>24</td><td> 8904135</td><td> 8973300</td><td>ENSDARP00000123402</td><td>                  </td><td>        </td><td>                  </td><td>          </td><td>      NA</td><td>      NA</td><td>             </td><td>                  </td><td>     NA</td><td>     NA</td><td>NA</td><td>    NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000045415</td><td>gnal    </td><td>24</td><td> 8904135</td><td> 8973300</td><td>ENSDARP00000066781</td><td>                  </td><td>        </td><td>                  </td><td>          </td><td>      NA</td><td>      NA</td><td>             </td><td>                  </td><td>     NA</td><td>     NA</td><td>NA</td><td>    NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000010509</td><td>        </td><td>ENSXETP00000023050</td><td>KV460380.1</td><td>  695815</td><td>  755758</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>39.1437</td><td>39.7516</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000038303</td><td>        </td><td>ENSXETP00000066651</td><td>KV460380.1</td><td>  585295</td><td>  593990</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>37.0031</td><td>38.0503</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000001428</td><td>        </td><td>ENSXETP00000003018</td><td>KV460380.1</td><td>  657394</td><td>  679990</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>37.0031</td><td>37.6947</td><td> 0</td><td> 90.99</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000035307</td><td>        </td><td>ENSXETP00000021357</td><td>KV460380.1</td><td>  636297</td><td>  637277</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>39.1437</td><td>39.2638</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000035638</td><td>        </td><td>ENSXETP00000003015</td><td>KV460380.1</td><td>  629793</td><td>  630770</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>39.4495</td><td>39.6923</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000036555</td><td>        </td><td>ENSXETP00000077786</td><td>KV460380.1</td><td>  516235</td><td>  568510</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>37.9205</td><td>36.9048</td><td> 0</td><td> 46.38</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000035336</td><td>ndufaf2 </td><td>ENSXETP00000056315</td><td>KV460380.1</td><td>  607003</td><td>  613853</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>37.0031</td><td>36.1194</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000034481</td><td>        </td><td>ENSXETP00000021354</td><td>KV460380.1</td><td>  585049</td><td>  586239</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>38.2263</td><td>38.8199</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000035889</td><td>        </td><td>ENSXETP00000066505</td><td>KV460380.1</td><td>  598407</td><td>  600057</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>35.4740</td><td>36.8254</td><td> 0</td><td> 90.59</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000101806</td><td>or104-1 </td><td>10</td><td>36354863</td><td>36356642</td><td>ENSDARP00000133585</td><td>ENSXETG00000027192</td><td>shisa2  </td><td>ENSXETP00000056884</td><td>KV460380.1</td><td>  763981</td><td>  765316</td><td>Euteleostomi </td><td>ortholog_many2many</td><td>32.7217</td><td>33.7539</td><td> 0</td><td>  0.00</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000039490</td><td>pitpnaa </td><td>15</td><td>28410664</td><td>28425011</td><td>ENSDARP00000122708</td><td>ENSXETG00000031018</td><td>pitpna  </td><td>ENSXETP00000085667</td><td>2         </td><td>39718101</td><td>39751925</td><td>Euteleostomi </td><td>ortholog_one2many </td><td>77.4908</td><td>72.6644</td><td>25</td><td> 92.07</td><td> 1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>ENSDARG00000112533</td><td>znf668        </td><td>CHR_ALT_CTG12_1_21</td><td>28958519</td><td>28960450</td><td>ENSDARP00000157662</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000110816</td><td>pou1f1        </td><td>CHR_ALT_CTG9_1_5  </td><td>19113370</td><td>19142350</td><td>ENSDARP00000155798</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000113590</td><td>slc2a12       </td><td>CHR_ALT_CTG23_1_34</td><td>31608711</td><td>31626232</td><td>ENSDARP00000135190</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000021789</td><td>myoc          </td><td>20                </td><td>15015557</td><td>15021628</td><td>ENSDARP00000028581</td><td>ENSXETG00000038908</td><td>myoc     </td><td>ENSXETP00000073502</td><td>4</td><td> 97460465</td><td> 97468737</td><td>Chordata     </td><td>ortholog_one2one </td><td>46.2025</td><td>43.8878</td><td>100</td><td>65.36</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000115266</td><td>imp3          </td><td>2                 </td><td> 9983079</td><td> 9989947</td><td>ENSDARP00000148639</td><td>ENSXETG00000038538</td><td>imp3     </td><td>ENSXETP00000100976</td><td>4</td><td>107643257</td><td>107708843</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>87.4317</td><td>87.4317</td><td> 25</td><td>85.87</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000115266</td><td>imp3          </td><td>2                 </td><td> 9983079</td><td> 9989947</td><td>ENSDARP00000154715</td><td>ENSXETG00000038538</td><td>imp3     </td><td>ENSXETP00000100976</td><td>4</td><td>107643257</td><td>107708843</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>87.4317</td><td>87.4317</td><td> 25</td><td>85.87</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000113557</td><td>fam167b       </td><td>CHR_ALT_CTG17_1_19</td><td>25392223</td><td>25396939</td><td>ENSDARP00000157436</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000101693</td><td>STK19         </td><td>KN150449.1        </td><td>   38186</td><td>   54173</td><td>ENSDARP00000135683</td><td>ENSXETG00000002497</td><td>stk19    </td><td>ENSXETP00000005317</td><td>8</td><td> 53693914</td><td> 53701289</td><td>Gnathostomata</td><td>ortholog_one2many</td><td>45.4902</td><td>44.6154</td><td>  0</td><td>19.19</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000035388</td><td>tmem174       </td><td>5                 </td><td>35105085</td><td>35108893</td><td>ENSDARP00000051296</td><td>ENSXETG00000037513</td><td>         </td><td>ENSXETP00000059943</td><td>1</td><td>166687146</td><td>166687814</td><td>Euteleostomi </td><td>ortholog_one2many</td><td>36.1345</td><td>38.7387</td><td>100</td><td>71.58</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000035388</td><td>tmem174       </td><td>5                 </td><td>35105085</td><td>35108893</td><td>ENSDARP00000051296</td><td>ENSXETG00000016816</td><td>tmem174.2</td><td>ENSXETP00000036664</td><td>1</td><td>166678058</td><td>166678894</td><td>Euteleostomi </td><td>ortholog_one2many</td><td>42.8571</td><td>36.6906</td><td>100</td><td>46.92</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000027824</td><td>top1mt        </td><td>24                </td><td>42074143</td><td>42075487</td><td>ENSDARP00000139898</td><td>ENSXETG00000004038</td><td>vapa     </td><td>ENSXETP00000008738</td><td>6</td><td> 89550547</td><td> 89570343</td><td>Gnathostomata</td><td>ortholog_one2one </td><td>62.0000</td><td>39.4068</td><td> 25</td><td>10.73</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000114110</td><td>rab8b         </td><td>CHR_ALT_CTG7_1_24 </td><td>29301641</td><td>29308455</td><td>ENSDARP00000151471</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000102167</td><td>ppox          </td><td>16                </td><td>42289076</td><td>42306597</td><td>ENSDARP00000135916</td><td>ENSXETG00000017673</td><td>ppox     </td><td>ENSXETP00000038355</td><td>8</td><td>117337070</td><td>117344793</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>55.3459</td><td>55.3459</td><td>  0</td><td>66.05</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000102167</td><td>ppox          </td><td>16                </td><td>42289076</td><td>42306597</td><td>ENSDARP00000149791</td><td>ENSXETG00000017673</td><td>ppox     </td><td>ENSXETP00000038355</td><td>8</td><td>117337070</td><td>117344793</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>55.3459</td><td>55.3459</td><td>  0</td><td>66.05</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000099092</td><td>RASGRP2       </td><td>7                 </td><td>72222542</td><td>72261721</td><td>ENSDARP00000136405</td><td>ENSXETG00000040253</td><td>rasgrp2  </td><td>ENSXETP00000094240</td><td>4</td><td> 43109116</td><td> 43140780</td><td>Euteleostomi </td><td>ortholog_one2many</td><td>61.3636</td><td>55.7659</td><td>  0</td><td>71.28</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000113927</td><td>eif4ebp1      </td><td>CHR_ALT_CTG21_1_12</td><td>21002912</td><td>21009442</td><td>ENSDARP00000147480</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000087697</td><td>lpl           </td><td>22                </td><td>15624371</td><td>15630421</td><td>ENSDARP00000109101</td><td>ENSXETG00000004467</td><td>lpl      </td><td>ENSXETP00000066597</td><td>1</td><td> 77715998</td><td> 77736810</td><td>Euteleostomi </td><td>ortholog_one2many</td><td>56.5558</td><td>59.4650</td><td>  0</td><td>70.31</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000099963</td><td>si:dkey-27n6.1</td><td>4                 </td><td>62057454</td><td>62061713</td><td>ENSDARP00000136664</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000116387</td><td>hapln3        </td><td>CHR_ALT_CTG25_1_12</td><td>19212058</td><td>19215380</td><td>ENSDARP00000146666</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000109998</td><td>ntn4          </td><td>CHR_ALT_CTG7_1_13 </td><td>19333456</td><td>19362387</td><td>ENSDARP00000135153</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000102333</td><td>fcf1          </td><td>17                </td><td>49457158</td><td>49481672</td><td>ENSDARP00000130801</td><td>ENSXETG00000027288</td><td>fcf1     </td><td>ENSXETP00000057007</td><td>8</td><td> 60956283</td><td> 60967007</td><td>Vertebrata   </td><td>ortholog_one2one </td><td>89.8990</td><td>90.3553</td><td> 25</td><td>87.83</td><td> 1</td></tr>\n",
       "\t<tr><td>ENSDARG00000099149</td><td>hs3st3b1a     </td><td>3                 </td><td>48842918</td><td>48869632</td><td>ENSDARP00000141379</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000103959</td><td>pak1          </td><td>18                </td><td> 3169579</td><td> 3286769</td><td>ENSDARP00000132462</td><td>ENSXETG00000010703</td><td>pak1     </td><td>ENSXETP00000025828</td><td>2</td><td>167246839</td><td>167282942</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>63.7138</td><td>71.7850</td><td> 25</td><td>18.82</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000103959</td><td>pak1          </td><td>18                </td><td> 3169579</td><td> 3286769</td><td>ENSDARP00000146047</td><td>ENSXETG00000010703</td><td>pak1     </td><td>ENSXETP00000025828</td><td>2</td><td>167246839</td><td>167282942</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>63.7138</td><td>71.7850</td><td> 25</td><td>18.82</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000103959</td><td>pak1          </td><td>18                </td><td> 3169579</td><td> 3286769</td><td>ENSDARP00000148256</td><td>ENSXETG00000010703</td><td>pak1     </td><td>ENSXETP00000025828</td><td>2</td><td>167246839</td><td>167282942</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>63.7138</td><td>71.7850</td><td> 25</td><td>18.82</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000103959</td><td>pak1          </td><td>18                </td><td> 3169579</td><td> 3286769</td><td>ENSDARP00000130006</td><td>ENSXETG00000010703</td><td>pak1     </td><td>ENSXETP00000025828</td><td>2</td><td>167246839</td><td>167282942</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>63.7138</td><td>71.7850</td><td> 25</td><td>18.82</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000103959</td><td>pak1          </td><td>18                </td><td> 3169579</td><td> 3286769</td><td>ENSDARP00000132912</td><td>ENSXETG00000010703</td><td>pak1     </td><td>ENSXETP00000025828</td><td>2</td><td>167246839</td><td>167282942</td><td>Euteleostomi </td><td>ortholog_one2one </td><td>63.7138</td><td>71.7850</td><td> 25</td><td>18.82</td><td> 0</td></tr>\n",
       "\t<tr><td>ENSDARG00000113512</td><td>ddhd1b        </td><td>CHR_ALT_CTG20_2_1 </td><td> 9787436</td><td> 9823395</td><td>ENSDARP00000156335</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000113512</td><td>ddhd1b        </td><td>CHR_ALT_CTG20_2_1 </td><td> 9787436</td><td> 9823395</td><td>ENSDARP00000145494</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "\t<tr><td>ENSDARG00000113512</td><td>ddhd1b        </td><td>CHR_ALT_CTG20_2_1 </td><td> 9787436</td><td> 9823395</td><td>ENSDARP00000151727</td><td>                  </td><td>         </td><td>                  </td><td> </td><td>       NA</td><td>       NA</td><td>             </td><td>                 </td><td>     NA</td><td>     NA</td><td> NA</td><td>   NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 43372 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " ensembl\\_gene\\_id & external\\_gene\\_name & chromosome\\_name & start\\_position & end\\_position & ensembl\\_peptide\\_id & xtropicalis\\_homolog\\_ensembl\\_gene & xtropicalis\\_homolog\\_associated\\_gene\\_name & xtropicalis\\_homolog\\_ensembl\\_peptide & xtropicalis\\_homolog\\_chromosome & xtropicalis\\_homolog\\_chrom\\_start & xtropicalis\\_homolog\\_chrom\\_end & xtropicalis\\_homolog\\_subtype & xtropicalis\\_homolog\\_orthology\\_type & xtropicalis\\_homolog\\_perc\\_id & xtropicalis\\_homolog\\_perc\\_id\\_r1 & xtropicalis\\_homolog\\_goc\\_score & xtropicalis\\_homolog\\_wga\\_coverage & xtropicalis\\_homolog\\_orthology\\_confidence\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <dbl> & <dbl> & <int> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSDARG00000063344 & fam162a  & 24 & 20927135 & 20934666 & ENSDARP00000123357 & ENSXETG00000026898 & fam162a  & ENSXETP00000056418 & KV460586.1 &      691 &     4121 & Euteleostomi  & ortholog\\_one2one   & 35.8974 & 47.8632 & 25 &  68.86 &  1\\\\\n",
       "\t ENSDARG00000063344 & fam162a  & 24 & 20927135 & 20934666 & ENSDARP00000114467 & ENSXETG00000026898 & fam162a  & ENSXETP00000056418 & KV460586.1 &      691 &     4121 & Euteleostomi  & ortholog\\_one2one   & 35.8974 & 47.8632 & 25 &  68.86 &  1\\\\\n",
       "\t ENSDARG00000103391 & rab9b    & 14 & 14603731 & 14607855 & ENSDARP00000140695 & ENSXETG00000018133 & rab9b    & ENSXETP00000039329 & 8          & 16957503 & 16961130 & Euteleostomi  & ortholog\\_one2one   & 75.2475 & 75.6219 & 25 & 100.00 &  1\\\\\n",
       "\t ENSDARG00000103391 & rab9b    & 14 & 14603731 & 14607855 & ENSDARP00000147450 & ENSXETG00000018133 & rab9b    & ENSXETP00000039329 & 8          & 16957503 & 16961130 & Euteleostomi  & ortholog\\_one2one   & 75.2475 & 75.6219 & 25 & 100.00 &  1\\\\\n",
       "\t ENSDARG00000080018 & kif16bb  & 13 & 34627550 & 34683370 & ENSDARP00000099496 &                    &          &                    &            &       NA &       NA &               &                    &      NA &      NA & NA &     NA & NA\\\\\n",
       "\t ENSDARG00000036109 & mtx1b    & 19 & 24448584 & 24484758 & ENSDARP00000052421 &                    &          &                    &            &       NA &       NA &               &                    &      NA &      NA & NA &     NA & NA\\\\\n",
       "\t ENSDARG00000075759 & samd4a   & 17 & 15033822 & 15117508 & ENSDARP00000129251 & ENSXETG00000012217 & samd4a   & ENSXETP00000073845 & 8          & 74362645 & 74457225 & Gnathostomata & ortholog\\_one2one   & 70.0581 & 67.6966 & 50 &  80.53 &  1\\\\\n",
       "\t ENSDARG00000075759 & samd4a   & 17 & 15033822 & 15117508 & ENSDARP00000101105 & ENSXETG00000012217 & samd4a   & ENSXETP00000073845 & 8          & 74362645 & 74457225 & Gnathostomata & ortholog\\_one2one   & 70.0581 & 67.6966 & 50 &  80.53 &  1\\\\\n",
       "\t ENSDARG00000015793 & creb3l1  & 7  & 38897836 & 38954874 & ENSDARP00000012411 & ENSXETG00000031075 & creb3l1  & ENSXETP00000062801 & 4          &  5396261 &  5434390 & Euteleostomi  & ortholog\\_one2one   & 51.9011 & 58.4582 & 75 &  60.05 &  1\\\\\n",
       "\t ENSDARG00000015793 & creb3l1  & 7  & 38897836 & 38954874 & ENSDARP00000142646 & ENSXETG00000031075 & creb3l1  & ENSXETP00000062801 & 4          &  5396261 &  5434390 & Euteleostomi  & ortholog\\_one2one   & 51.9011 & 58.4582 & 75 &  60.05 &  1\\\\\n",
       "\t ENSDARG00000015793 & creb3l1  & 7  & 38897836 & 38954874 & ENSDARP00000134652 & ENSXETG00000031075 & creb3l1  & ENSXETP00000062801 & 4          &  5396261 &  5434390 & Euteleostomi  & ortholog\\_one2one   & 51.9011 & 58.4582 & 75 &  60.05 &  1\\\\\n",
       "\t ENSDARG00000044573 & cdc42    & 11 & 39120306 & 39148199 & ENSDARP00000129339 & ENSXETG00000004743 & cdc42    & ENSXETP00000030366 & 7          & 68594070 & 68619099 & Chordata      & ortholog\\_one2one   & 98.4293 & 88.6792 & 75 &  94.10 &  1\\\\\n",
       "\t ENSDARG00000044573 & cdc42    & 11 & 39120306 & 39148199 & ENSDARP00000129719 & ENSXETG00000004743 & cdc42    & ENSXETP00000030366 & 7          & 68594070 & 68619099 & Chordata      & ortholog\\_one2one   & 98.4293 & 88.6792 & 75 &  94.10 &  1\\\\\n",
       "\t ENSDARG00000044573 & cdc42    & 11 & 39120306 & 39148199 & ENSDARP00000149720 & ENSXETG00000004743 & cdc42    & ENSXETP00000030366 & 7          & 68594070 & 68619099 & Chordata      & ortholog\\_one2one   & 98.4293 & 88.6792 & 75 &  94.10 &  1\\\\\n",
       "\t ENSDARG00000044573 & cdc42    & 11 & 39120306 & 39148199 & ENSDARP00000155235 & ENSXETG00000004743 & cdc42    & ENSXETP00000030366 & 7          & 68594070 & 68619099 & Chordata      & ortholog\\_one2one   & 98.4293 & 88.6792 & 75 &  94.10 &  1\\\\\n",
       "\t ENSDARG00000076270 & adamts13 & 8  & 45226586 & 45279411 & ENSDARP00000119445 & ENSXETG00000006882 & adamts13 & ENSXETP00000070357 & KV460557.1 &    60725 &    78242 & Euteleostomi  & ortholog\\_one2one   & 38.6515 & 42.8889 &  0 &  35.79 &  0\\\\\n",
       "\t ENSDARG00000076270 & adamts13 & 8  & 45226586 & 45279411 & ENSDARP00000143739 & ENSXETG00000006882 & adamts13 & ENSXETP00000070357 & KV460557.1 &    60725 &    78242 & Euteleostomi  & ortholog\\_one2one   & 38.6515 & 42.8889 &  0 &  35.79 &  0\\\\\n",
       "\t ENSDARG00000045415 & gnal     & 24 &  8904135 &  8973300 & ENSDARP00000123402 &                    &          &                    &            &       NA &       NA &               &                    &      NA &      NA & NA &     NA & NA\\\\\n",
       "\t ENSDARG00000045415 & gnal     & 24 &  8904135 &  8973300 & ENSDARP00000066781 &                    &          &                    &            &       NA &       NA &               &                    &      NA &      NA & NA &     NA & NA\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000010509 &          & ENSXETP00000023050 & KV460380.1 &   695815 &   755758 & Euteleostomi  & ortholog\\_many2many & 39.1437 & 39.7516 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000038303 &          & ENSXETP00000066651 & KV460380.1 &   585295 &   593990 & Euteleostomi  & ortholog\\_many2many & 37.0031 & 38.0503 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000001428 &          & ENSXETP00000003018 & KV460380.1 &   657394 &   679990 & Euteleostomi  & ortholog\\_many2many & 37.0031 & 37.6947 &  0 &  90.99 &  1\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000035307 &          & ENSXETP00000021357 & KV460380.1 &   636297 &   637277 & Euteleostomi  & ortholog\\_many2many & 39.1437 & 39.2638 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000035638 &          & ENSXETP00000003015 & KV460380.1 &   629793 &   630770 & Euteleostomi  & ortholog\\_many2many & 39.4495 & 39.6923 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000036555 &          & ENSXETP00000077786 & KV460380.1 &   516235 &   568510 & Euteleostomi  & ortholog\\_many2many & 37.9205 & 36.9048 &  0 &  46.38 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000035336 & ndufaf2  & ENSXETP00000056315 & KV460380.1 &   607003 &   613853 & Euteleostomi  & ortholog\\_many2many & 37.0031 & 36.1194 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000034481 &          & ENSXETP00000021354 & KV460380.1 &   585049 &   586239 & Euteleostomi  & ortholog\\_many2many & 38.2263 & 38.8199 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000035889 &          & ENSXETP00000066505 & KV460380.1 &   598407 &   600057 & Euteleostomi  & ortholog\\_many2many & 35.4740 & 36.8254 &  0 &  90.59 &  1\\\\\n",
       "\t ENSDARG00000101806 & or104-1  & 10 & 36354863 & 36356642 & ENSDARP00000133585 & ENSXETG00000027192 & shisa2   & ENSXETP00000056884 & KV460380.1 &   763981 &   765316 & Euteleostomi  & ortholog\\_many2many & 32.7217 & 33.7539 &  0 &   0.00 &  0\\\\\n",
       "\t ENSDARG00000039490 & pitpnaa  & 15 & 28410664 & 28425011 & ENSDARP00000122708 & ENSXETG00000031018 & pitpna   & ENSXETP00000085667 & 2          & 39718101 & 39751925 & Euteleostomi  & ortholog\\_one2many  & 77.4908 & 72.6644 & 25 &  92.07 &  1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t ENSDARG00000112533 & znf668         & CHR\\_ALT\\_CTG12\\_1\\_21 & 28958519 & 28960450 & ENSDARP00000157662 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000110816 & pou1f1         & CHR\\_ALT\\_CTG9\\_1\\_5   & 19113370 & 19142350 & ENSDARP00000155798 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000113590 & slc2a12        & CHR\\_ALT\\_CTG23\\_1\\_34 & 31608711 & 31626232 & ENSDARP00000135190 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000021789 & myoc           & 20                 & 15015557 & 15021628 & ENSDARP00000028581 & ENSXETG00000038908 & myoc      & ENSXETP00000073502 & 4 &  97460465 &  97468737 & Chordata      & ortholog\\_one2one  & 46.2025 & 43.8878 & 100 & 65.36 &  1\\\\\n",
       "\t ENSDARG00000115266 & imp3           & 2                  &  9983079 &  9989947 & ENSDARP00000148639 & ENSXETG00000038538 & imp3      & ENSXETP00000100976 & 4 & 107643257 & 107708843 & Euteleostomi  & ortholog\\_one2one  & 87.4317 & 87.4317 &  25 & 85.87 &  1\\\\\n",
       "\t ENSDARG00000115266 & imp3           & 2                  &  9983079 &  9989947 & ENSDARP00000154715 & ENSXETG00000038538 & imp3      & ENSXETP00000100976 & 4 & 107643257 & 107708843 & Euteleostomi  & ortholog\\_one2one  & 87.4317 & 87.4317 &  25 & 85.87 &  1\\\\\n",
       "\t ENSDARG00000113557 & fam167b        & CHR\\_ALT\\_CTG17\\_1\\_19 & 25392223 & 25396939 & ENSDARP00000157436 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000101693 & STK19          & KN150449.1         &    38186 &    54173 & ENSDARP00000135683 & ENSXETG00000002497 & stk19     & ENSXETP00000005317 & 8 &  53693914 &  53701289 & Gnathostomata & ortholog\\_one2many & 45.4902 & 44.6154 &   0 & 19.19 &  0\\\\\n",
       "\t ENSDARG00000035388 & tmem174        & 5                  & 35105085 & 35108893 & ENSDARP00000051296 & ENSXETG00000037513 &           & ENSXETP00000059943 & 1 & 166687146 & 166687814 & Euteleostomi  & ortholog\\_one2many & 36.1345 & 38.7387 & 100 & 71.58 &  1\\\\\n",
       "\t ENSDARG00000035388 & tmem174        & 5                  & 35105085 & 35108893 & ENSDARP00000051296 & ENSXETG00000016816 & tmem174.2 & ENSXETP00000036664 & 1 & 166678058 & 166678894 & Euteleostomi  & ortholog\\_one2many & 42.8571 & 36.6906 & 100 & 46.92 &  1\\\\\n",
       "\t ENSDARG00000027824 & top1mt         & 24                 & 42074143 & 42075487 & ENSDARP00000139898 & ENSXETG00000004038 & vapa      & ENSXETP00000008738 & 6 &  89550547 &  89570343 & Gnathostomata & ortholog\\_one2one  & 62.0000 & 39.4068 &  25 & 10.73 &  0\\\\\n",
       "\t ENSDARG00000114110 & rab8b          & CHR\\_ALT\\_CTG7\\_1\\_24  & 29301641 & 29308455 & ENSDARP00000151471 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000102167 & ppox           & 16                 & 42289076 & 42306597 & ENSDARP00000135916 & ENSXETG00000017673 & ppox      & ENSXETP00000038355 & 8 & 117337070 & 117344793 & Euteleostomi  & ortholog\\_one2one  & 55.3459 & 55.3459 &   0 & 66.05 &  1\\\\\n",
       "\t ENSDARG00000102167 & ppox           & 16                 & 42289076 & 42306597 & ENSDARP00000149791 & ENSXETG00000017673 & ppox      & ENSXETP00000038355 & 8 & 117337070 & 117344793 & Euteleostomi  & ortholog\\_one2one  & 55.3459 & 55.3459 &   0 & 66.05 &  1\\\\\n",
       "\t ENSDARG00000099092 & RASGRP2        & 7                  & 72222542 & 72261721 & ENSDARP00000136405 & ENSXETG00000040253 & rasgrp2   & ENSXETP00000094240 & 4 &  43109116 &  43140780 & Euteleostomi  & ortholog\\_one2many & 61.3636 & 55.7659 &   0 & 71.28 &  1\\\\\n",
       "\t ENSDARG00000113927 & eif4ebp1       & CHR\\_ALT\\_CTG21\\_1\\_12 & 21002912 & 21009442 & ENSDARP00000147480 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000087697 & lpl            & 22                 & 15624371 & 15630421 & ENSDARP00000109101 & ENSXETG00000004467 & lpl       & ENSXETP00000066597 & 1 &  77715998 &  77736810 & Euteleostomi  & ortholog\\_one2many & 56.5558 & 59.4650 &   0 & 70.31 &  1\\\\\n",
       "\t ENSDARG00000099963 & si:dkey-27n6.1 & 4                  & 62057454 & 62061713 & ENSDARP00000136664 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000116387 & hapln3         & CHR\\_ALT\\_CTG25\\_1\\_12 & 19212058 & 19215380 & ENSDARP00000146666 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000109998 & ntn4           & CHR\\_ALT\\_CTG7\\_1\\_13  & 19333456 & 19362387 & ENSDARP00000135153 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000102333 & fcf1           & 17                 & 49457158 & 49481672 & ENSDARP00000130801 & ENSXETG00000027288 & fcf1      & ENSXETP00000057007 & 8 &  60956283 &  60967007 & Vertebrata    & ortholog\\_one2one  & 89.8990 & 90.3553 &  25 & 87.83 &  1\\\\\n",
       "\t ENSDARG00000099149 & hs3st3b1a      & 3                  & 48842918 & 48869632 & ENSDARP00000141379 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000103959 & pak1           & 18                 &  3169579 &  3286769 & ENSDARP00000132462 & ENSXETG00000010703 & pak1      & ENSXETP00000025828 & 2 & 167246839 & 167282942 & Euteleostomi  & ortholog\\_one2one  & 63.7138 & 71.7850 &  25 & 18.82 &  0\\\\\n",
       "\t ENSDARG00000103959 & pak1           & 18                 &  3169579 &  3286769 & ENSDARP00000146047 & ENSXETG00000010703 & pak1      & ENSXETP00000025828 & 2 & 167246839 & 167282942 & Euteleostomi  & ortholog\\_one2one  & 63.7138 & 71.7850 &  25 & 18.82 &  0\\\\\n",
       "\t ENSDARG00000103959 & pak1           & 18                 &  3169579 &  3286769 & ENSDARP00000148256 & ENSXETG00000010703 & pak1      & ENSXETP00000025828 & 2 & 167246839 & 167282942 & Euteleostomi  & ortholog\\_one2one  & 63.7138 & 71.7850 &  25 & 18.82 &  0\\\\\n",
       "\t ENSDARG00000103959 & pak1           & 18                 &  3169579 &  3286769 & ENSDARP00000130006 & ENSXETG00000010703 & pak1      & ENSXETP00000025828 & 2 & 167246839 & 167282942 & Euteleostomi  & ortholog\\_one2one  & 63.7138 & 71.7850 &  25 & 18.82 &  0\\\\\n",
       "\t ENSDARG00000103959 & pak1           & 18                 &  3169579 &  3286769 & ENSDARP00000132912 & ENSXETG00000010703 & pak1      & ENSXETP00000025828 & 2 & 167246839 & 167282942 & Euteleostomi  & ortholog\\_one2one  & 63.7138 & 71.7850 &  25 & 18.82 &  0\\\\\n",
       "\t ENSDARG00000113512 & ddhd1b         & CHR\\_ALT\\_CTG20\\_2\\_1  &  9787436 &  9823395 & ENSDARP00000156335 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000113512 & ddhd1b         & CHR\\_ALT\\_CTG20\\_2\\_1  &  9787436 &  9823395 & ENSDARP00000145494 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\t ENSDARG00000113512 & ddhd1b         & CHR\\_ALT\\_CTG20\\_2\\_1  &  9787436 &  9823395 & ENSDARP00000151727 &                    &           &                    &   &        NA &        NA &               &                   &      NA &      NA &  NA &    NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 43372 × 19\n",
       "\n",
       "| ensembl_gene_id &lt;chr&gt; | external_gene_name &lt;chr&gt; | chromosome_name &lt;chr&gt; | start_position &lt;int&gt; | end_position &lt;int&gt; | ensembl_peptide_id &lt;chr&gt; | xtropicalis_homolog_ensembl_gene &lt;chr&gt; | xtropicalis_homolog_associated_gene_name &lt;chr&gt; | xtropicalis_homolog_ensembl_peptide &lt;chr&gt; | xtropicalis_homolog_chromosome &lt;chr&gt; | xtropicalis_homolog_chrom_start &lt;int&gt; | xtropicalis_homolog_chrom_end &lt;int&gt; | xtropicalis_homolog_subtype &lt;chr&gt; | xtropicalis_homolog_orthology_type &lt;chr&gt; | xtropicalis_homolog_perc_id &lt;dbl&gt; | xtropicalis_homolog_perc_id_r1 &lt;dbl&gt; | xtropicalis_homolog_goc_score &lt;int&gt; | xtropicalis_homolog_wga_coverage &lt;dbl&gt; | xtropicalis_homolog_orthology_confidence &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSDARG00000063344 | fam162a  | 24 | 20927135 | 20934666 | ENSDARP00000123357 | ENSXETG00000026898 | fam162a  | ENSXETP00000056418 | KV460586.1 |      691 |     4121 | Euteleostomi  | ortholog_one2one   | 35.8974 | 47.8632 | 25 |  68.86 |  1 |\n",
       "| ENSDARG00000063344 | fam162a  | 24 | 20927135 | 20934666 | ENSDARP00000114467 | ENSXETG00000026898 | fam162a  | ENSXETP00000056418 | KV460586.1 |      691 |     4121 | Euteleostomi  | ortholog_one2one   | 35.8974 | 47.8632 | 25 |  68.86 |  1 |\n",
       "| ENSDARG00000103391 | rab9b    | 14 | 14603731 | 14607855 | ENSDARP00000140695 | ENSXETG00000018133 | rab9b    | ENSXETP00000039329 | 8          | 16957503 | 16961130 | Euteleostomi  | ortholog_one2one   | 75.2475 | 75.6219 | 25 | 100.00 |  1 |\n",
       "| ENSDARG00000103391 | rab9b    | 14 | 14603731 | 14607855 | ENSDARP00000147450 | ENSXETG00000018133 | rab9b    | ENSXETP00000039329 | 8          | 16957503 | 16961130 | Euteleostomi  | ortholog_one2one   | 75.2475 | 75.6219 | 25 | 100.00 |  1 |\n",
       "| ENSDARG00000080018 | kif16bb  | 13 | 34627550 | 34683370 | ENSDARP00000099496 | <!----> | <!----> | <!----> | <!----> |       NA |       NA | <!----> | <!----> |      NA |      NA | NA |     NA | NA |\n",
       "| ENSDARG00000036109 | mtx1b    | 19 | 24448584 | 24484758 | ENSDARP00000052421 | <!----> | <!----> | <!----> | <!----> |       NA |       NA | <!----> | <!----> |      NA |      NA | NA |     NA | NA |\n",
       "| ENSDARG00000075759 | samd4a   | 17 | 15033822 | 15117508 | ENSDARP00000129251 | ENSXETG00000012217 | samd4a   | ENSXETP00000073845 | 8          | 74362645 | 74457225 | Gnathostomata | ortholog_one2one   | 70.0581 | 67.6966 | 50 |  80.53 |  1 |\n",
       "| ENSDARG00000075759 | samd4a   | 17 | 15033822 | 15117508 | ENSDARP00000101105 | ENSXETG00000012217 | samd4a   | ENSXETP00000073845 | 8          | 74362645 | 74457225 | Gnathostomata | ortholog_one2one   | 70.0581 | 67.6966 | 50 |  80.53 |  1 |\n",
       "| ENSDARG00000015793 | creb3l1  | 7  | 38897836 | 38954874 | ENSDARP00000012411 | ENSXETG00000031075 | creb3l1  | ENSXETP00000062801 | 4          |  5396261 |  5434390 | Euteleostomi  | ortholog_one2one   | 51.9011 | 58.4582 | 75 |  60.05 |  1 |\n",
       "| ENSDARG00000015793 | creb3l1  | 7  | 38897836 | 38954874 | ENSDARP00000142646 | ENSXETG00000031075 | creb3l1  | ENSXETP00000062801 | 4          |  5396261 |  5434390 | Euteleostomi  | ortholog_one2one   | 51.9011 | 58.4582 | 75 |  60.05 |  1 |\n",
       "| ENSDARG00000015793 | creb3l1  | 7  | 38897836 | 38954874 | ENSDARP00000134652 | ENSXETG00000031075 | creb3l1  | ENSXETP00000062801 | 4          |  5396261 |  5434390 | Euteleostomi  | ortholog_one2one   | 51.9011 | 58.4582 | 75 |  60.05 |  1 |\n",
       "| ENSDARG00000044573 | cdc42    | 11 | 39120306 | 39148199 | ENSDARP00000129339 | ENSXETG00000004743 | cdc42    | ENSXETP00000030366 | 7          | 68594070 | 68619099 | Chordata      | ortholog_one2one   | 98.4293 | 88.6792 | 75 |  94.10 |  1 |\n",
       "| ENSDARG00000044573 | cdc42    | 11 | 39120306 | 39148199 | ENSDARP00000129719 | ENSXETG00000004743 | cdc42    | ENSXETP00000030366 | 7          | 68594070 | 68619099 | Chordata      | ortholog_one2one   | 98.4293 | 88.6792 | 75 |  94.10 |  1 |\n",
       "| ENSDARG00000044573 | cdc42    | 11 | 39120306 | 39148199 | ENSDARP00000149720 | ENSXETG00000004743 | cdc42    | ENSXETP00000030366 | 7          | 68594070 | 68619099 | Chordata      | ortholog_one2one   | 98.4293 | 88.6792 | 75 |  94.10 |  1 |\n",
       "| ENSDARG00000044573 | cdc42    | 11 | 39120306 | 39148199 | ENSDARP00000155235 | ENSXETG00000004743 | cdc42    | ENSXETP00000030366 | 7          | 68594070 | 68619099 | Chordata      | ortholog_one2one   | 98.4293 | 88.6792 | 75 |  94.10 |  1 |\n",
       "| ENSDARG00000076270 | adamts13 | 8  | 45226586 | 45279411 | ENSDARP00000119445 | ENSXETG00000006882 | adamts13 | ENSXETP00000070357 | KV460557.1 |    60725 |    78242 | Euteleostomi  | ortholog_one2one   | 38.6515 | 42.8889 |  0 |  35.79 |  0 |\n",
       "| ENSDARG00000076270 | adamts13 | 8  | 45226586 | 45279411 | ENSDARP00000143739 | ENSXETG00000006882 | adamts13 | ENSXETP00000070357 | KV460557.1 |    60725 |    78242 | Euteleostomi  | ortholog_one2one   | 38.6515 | 42.8889 |  0 |  35.79 |  0 |\n",
       "| ENSDARG00000045415 | gnal     | 24 |  8904135 |  8973300 | ENSDARP00000123402 | <!----> | <!----> | <!----> | <!----> |       NA |       NA | <!----> | <!----> |      NA |      NA | NA |     NA | NA |\n",
       "| ENSDARG00000045415 | gnal     | 24 |  8904135 |  8973300 | ENSDARP00000066781 | <!----> | <!----> | <!----> | <!----> |       NA |       NA | <!----> | <!----> |      NA |      NA | NA |     NA | NA |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000010509 | <!----> | ENSXETP00000023050 | KV460380.1 |   695815 |   755758 | Euteleostomi  | ortholog_many2many | 39.1437 | 39.7516 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000038303 | <!----> | ENSXETP00000066651 | KV460380.1 |   585295 |   593990 | Euteleostomi  | ortholog_many2many | 37.0031 | 38.0503 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000001428 | <!----> | ENSXETP00000003018 | KV460380.1 |   657394 |   679990 | Euteleostomi  | ortholog_many2many | 37.0031 | 37.6947 |  0 |  90.99 |  1 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000035307 | <!----> | ENSXETP00000021357 | KV460380.1 |   636297 |   637277 | Euteleostomi  | ortholog_many2many | 39.1437 | 39.2638 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000035638 | <!----> | ENSXETP00000003015 | KV460380.1 |   629793 |   630770 | Euteleostomi  | ortholog_many2many | 39.4495 | 39.6923 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000036555 | <!----> | ENSXETP00000077786 | KV460380.1 |   516235 |   568510 | Euteleostomi  | ortholog_many2many | 37.9205 | 36.9048 |  0 |  46.38 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000035336 | ndufaf2  | ENSXETP00000056315 | KV460380.1 |   607003 |   613853 | Euteleostomi  | ortholog_many2many | 37.0031 | 36.1194 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000034481 | <!----> | ENSXETP00000021354 | KV460380.1 |   585049 |   586239 | Euteleostomi  | ortholog_many2many | 38.2263 | 38.8199 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000035889 | <!----> | ENSXETP00000066505 | KV460380.1 |   598407 |   600057 | Euteleostomi  | ortholog_many2many | 35.4740 | 36.8254 |  0 |  90.59 |  1 |\n",
       "| ENSDARG00000101806 | or104-1  | 10 | 36354863 | 36356642 | ENSDARP00000133585 | ENSXETG00000027192 | shisa2   | ENSXETP00000056884 | KV460380.1 |   763981 |   765316 | Euteleostomi  | ortholog_many2many | 32.7217 | 33.7539 |  0 |   0.00 |  0 |\n",
       "| ENSDARG00000039490 | pitpnaa  | 15 | 28410664 | 28425011 | ENSDARP00000122708 | ENSXETG00000031018 | pitpna   | ENSXETP00000085667 | 2          | 39718101 | 39751925 | Euteleostomi  | ortholog_one2many  | 77.4908 | 72.6644 | 25 |  92.07 |  1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| ENSDARG00000112533 | znf668         | CHR_ALT_CTG12_1_21 | 28958519 | 28960450 | ENSDARP00000157662 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000110816 | pou1f1         | CHR_ALT_CTG9_1_5   | 19113370 | 19142350 | ENSDARP00000155798 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000113590 | slc2a12        | CHR_ALT_CTG23_1_34 | 31608711 | 31626232 | ENSDARP00000135190 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000021789 | myoc           | 20                 | 15015557 | 15021628 | ENSDARP00000028581 | ENSXETG00000038908 | myoc      | ENSXETP00000073502 | 4 |  97460465 |  97468737 | Chordata      | ortholog_one2one  | 46.2025 | 43.8878 | 100 | 65.36 |  1 |\n",
       "| ENSDARG00000115266 | imp3           | 2                  |  9983079 |  9989947 | ENSDARP00000148639 | ENSXETG00000038538 | imp3      | ENSXETP00000100976 | 4 | 107643257 | 107708843 | Euteleostomi  | ortholog_one2one  | 87.4317 | 87.4317 |  25 | 85.87 |  1 |\n",
       "| ENSDARG00000115266 | imp3           | 2                  |  9983079 |  9989947 | ENSDARP00000154715 | ENSXETG00000038538 | imp3      | ENSXETP00000100976 | 4 | 107643257 | 107708843 | Euteleostomi  | ortholog_one2one  | 87.4317 | 87.4317 |  25 | 85.87 |  1 |\n",
       "| ENSDARG00000113557 | fam167b        | CHR_ALT_CTG17_1_19 | 25392223 | 25396939 | ENSDARP00000157436 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000101693 | STK19          | KN150449.1         |    38186 |    54173 | ENSDARP00000135683 | ENSXETG00000002497 | stk19     | ENSXETP00000005317 | 8 |  53693914 |  53701289 | Gnathostomata | ortholog_one2many | 45.4902 | 44.6154 |   0 | 19.19 |  0 |\n",
       "| ENSDARG00000035388 | tmem174        | 5                  | 35105085 | 35108893 | ENSDARP00000051296 | ENSXETG00000037513 | <!----> | ENSXETP00000059943 | 1 | 166687146 | 166687814 | Euteleostomi  | ortholog_one2many | 36.1345 | 38.7387 | 100 | 71.58 |  1 |\n",
       "| ENSDARG00000035388 | tmem174        | 5                  | 35105085 | 35108893 | ENSDARP00000051296 | ENSXETG00000016816 | tmem174.2 | ENSXETP00000036664 | 1 | 166678058 | 166678894 | Euteleostomi  | ortholog_one2many | 42.8571 | 36.6906 | 100 | 46.92 |  1 |\n",
       "| ENSDARG00000027824 | top1mt         | 24                 | 42074143 | 42075487 | ENSDARP00000139898 | ENSXETG00000004038 | vapa      | ENSXETP00000008738 | 6 |  89550547 |  89570343 | Gnathostomata | ortholog_one2one  | 62.0000 | 39.4068 |  25 | 10.73 |  0 |\n",
       "| ENSDARG00000114110 | rab8b          | CHR_ALT_CTG7_1_24  | 29301641 | 29308455 | ENSDARP00000151471 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000102167 | ppox           | 16                 | 42289076 | 42306597 | ENSDARP00000135916 | ENSXETG00000017673 | ppox      | ENSXETP00000038355 | 8 | 117337070 | 117344793 | Euteleostomi  | ortholog_one2one  | 55.3459 | 55.3459 |   0 | 66.05 |  1 |\n",
       "| ENSDARG00000102167 | ppox           | 16                 | 42289076 | 42306597 | ENSDARP00000149791 | ENSXETG00000017673 | ppox      | ENSXETP00000038355 | 8 | 117337070 | 117344793 | Euteleostomi  | ortholog_one2one  | 55.3459 | 55.3459 |   0 | 66.05 |  1 |\n",
       "| ENSDARG00000099092 | RASGRP2        | 7                  | 72222542 | 72261721 | ENSDARP00000136405 | ENSXETG00000040253 | rasgrp2   | ENSXETP00000094240 | 4 |  43109116 |  43140780 | Euteleostomi  | ortholog_one2many | 61.3636 | 55.7659 |   0 | 71.28 |  1 |\n",
       "| ENSDARG00000113927 | eif4ebp1       | CHR_ALT_CTG21_1_12 | 21002912 | 21009442 | ENSDARP00000147480 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000087697 | lpl            | 22                 | 15624371 | 15630421 | ENSDARP00000109101 | ENSXETG00000004467 | lpl       | ENSXETP00000066597 | 1 |  77715998 |  77736810 | Euteleostomi  | ortholog_one2many | 56.5558 | 59.4650 |   0 | 70.31 |  1 |\n",
       "| ENSDARG00000099963 | si:dkey-27n6.1 | 4                  | 62057454 | 62061713 | ENSDARP00000136664 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000116387 | hapln3         | CHR_ALT_CTG25_1_12 | 19212058 | 19215380 | ENSDARP00000146666 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000109998 | ntn4           | CHR_ALT_CTG7_1_13  | 19333456 | 19362387 | ENSDARP00000135153 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000102333 | fcf1           | 17                 | 49457158 | 49481672 | ENSDARP00000130801 | ENSXETG00000027288 | fcf1      | ENSXETP00000057007 | 8 |  60956283 |  60967007 | Vertebrata    | ortholog_one2one  | 89.8990 | 90.3553 |  25 | 87.83 |  1 |\n",
       "| ENSDARG00000099149 | hs3st3b1a      | 3                  | 48842918 | 48869632 | ENSDARP00000141379 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000103959 | pak1           | 18                 |  3169579 |  3286769 | ENSDARP00000132462 | ENSXETG00000010703 | pak1      | ENSXETP00000025828 | 2 | 167246839 | 167282942 | Euteleostomi  | ortholog_one2one  | 63.7138 | 71.7850 |  25 | 18.82 |  0 |\n",
       "| ENSDARG00000103959 | pak1           | 18                 |  3169579 |  3286769 | ENSDARP00000146047 | ENSXETG00000010703 | pak1      | ENSXETP00000025828 | 2 | 167246839 | 167282942 | Euteleostomi  | ortholog_one2one  | 63.7138 | 71.7850 |  25 | 18.82 |  0 |\n",
       "| ENSDARG00000103959 | pak1           | 18                 |  3169579 |  3286769 | ENSDARP00000148256 | ENSXETG00000010703 | pak1      | ENSXETP00000025828 | 2 | 167246839 | 167282942 | Euteleostomi  | ortholog_one2one  | 63.7138 | 71.7850 |  25 | 18.82 |  0 |\n",
       "| ENSDARG00000103959 | pak1           | 18                 |  3169579 |  3286769 | ENSDARP00000130006 | ENSXETG00000010703 | pak1      | ENSXETP00000025828 | 2 | 167246839 | 167282942 | Euteleostomi  | ortholog_one2one  | 63.7138 | 71.7850 |  25 | 18.82 |  0 |\n",
       "| ENSDARG00000103959 | pak1           | 18                 |  3169579 |  3286769 | ENSDARP00000132912 | ENSXETG00000010703 | pak1      | ENSXETP00000025828 | 2 | 167246839 | 167282942 | Euteleostomi  | ortholog_one2one  | 63.7138 | 71.7850 |  25 | 18.82 |  0 |\n",
       "| ENSDARG00000113512 | ddhd1b         | CHR_ALT_CTG20_2_1  |  9787436 |  9823395 | ENSDARP00000156335 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000113512 | ddhd1b         | CHR_ALT_CTG20_2_1  |  9787436 |  9823395 | ENSDARP00000145494 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "| ENSDARG00000113512 | ddhd1b         | CHR_ALT_CTG20_2_1  |  9787436 |  9823395 | ENSDARP00000151727 | <!----> | <!----> | <!----> | <!----> |        NA |        NA | <!----> | <!----> |      NA |      NA |  NA |    NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "      ensembl_gene_id    external_gene_name chromosome_name    start_position\n",
       "1     ENSDARG00000063344 fam162a            24                 20927135      \n",
       "2     ENSDARG00000063344 fam162a            24                 20927135      \n",
       "3     ENSDARG00000103391 rab9b              14                 14603731      \n",
       "4     ENSDARG00000103391 rab9b              14                 14603731      \n",
       "5     ENSDARG00000080018 kif16bb            13                 34627550      \n",
       "6     ENSDARG00000036109 mtx1b              19                 24448584      \n",
       "7     ENSDARG00000075759 samd4a             17                 15033822      \n",
       "8     ENSDARG00000075759 samd4a             17                 15033822      \n",
       "9     ENSDARG00000015793 creb3l1            7                  38897836      \n",
       "10    ENSDARG00000015793 creb3l1            7                  38897836      \n",
       "11    ENSDARG00000015793 creb3l1            7                  38897836      \n",
       "12    ENSDARG00000044573 cdc42              11                 39120306      \n",
       "13    ENSDARG00000044573 cdc42              11                 39120306      \n",
       "14    ENSDARG00000044573 cdc42              11                 39120306      \n",
       "15    ENSDARG00000044573 cdc42              11                 39120306      \n",
       "16    ENSDARG00000076270 adamts13           8                  45226586      \n",
       "17    ENSDARG00000076270 adamts13           8                  45226586      \n",
       "18    ENSDARG00000045415 gnal               24                  8904135      \n",
       "19    ENSDARG00000045415 gnal               24                  8904135      \n",
       "20    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "21    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "22    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "23    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "24    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "25    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "26    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "27    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "28    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "29    ENSDARG00000101806 or104-1            10                 36354863      \n",
       "30    ENSDARG00000039490 pitpnaa            15                 28410664      \n",
       "⋮     ⋮                  ⋮                  ⋮                  ⋮             \n",
       "43343 ENSDARG00000112533 znf668             CHR_ALT_CTG12_1_21 28958519      \n",
       "43344 ENSDARG00000110816 pou1f1             CHR_ALT_CTG9_1_5   19113370      \n",
       "43345 ENSDARG00000113590 slc2a12            CHR_ALT_CTG23_1_34 31608711      \n",
       "43346 ENSDARG00000021789 myoc               20                 15015557      \n",
       "43347 ENSDARG00000115266 imp3               2                   9983079      \n",
       "43348 ENSDARG00000115266 imp3               2                   9983079      \n",
       "43349 ENSDARG00000113557 fam167b            CHR_ALT_CTG17_1_19 25392223      \n",
       "43350 ENSDARG00000101693 STK19              KN150449.1            38186      \n",
       "43351 ENSDARG00000035388 tmem174            5                  35105085      \n",
       "43352 ENSDARG00000035388 tmem174            5                  35105085      \n",
       "43353 ENSDARG00000027824 top1mt             24                 42074143      \n",
       "43354 ENSDARG00000114110 rab8b              CHR_ALT_CTG7_1_24  29301641      \n",
       "43355 ENSDARG00000102167 ppox               16                 42289076      \n",
       "43356 ENSDARG00000102167 ppox               16                 42289076      \n",
       "43357 ENSDARG00000099092 RASGRP2            7                  72222542      \n",
       "43358 ENSDARG00000113927 eif4ebp1           CHR_ALT_CTG21_1_12 21002912      \n",
       "43359 ENSDARG00000087697 lpl                22                 15624371      \n",
       "43360 ENSDARG00000099963 si:dkey-27n6.1     4                  62057454      \n",
       "43361 ENSDARG00000116387 hapln3             CHR_ALT_CTG25_1_12 19212058      \n",
       "43362 ENSDARG00000109998 ntn4               CHR_ALT_CTG7_1_13  19333456      \n",
       "43363 ENSDARG00000102333 fcf1               17                 49457158      \n",
       "43364 ENSDARG00000099149 hs3st3b1a          3                  48842918      \n",
       "43365 ENSDARG00000103959 pak1               18                  3169579      \n",
       "43366 ENSDARG00000103959 pak1               18                  3169579      \n",
       "43367 ENSDARG00000103959 pak1               18                  3169579      \n",
       "43368 ENSDARG00000103959 pak1               18                  3169579      \n",
       "43369 ENSDARG00000103959 pak1               18                  3169579      \n",
       "43370 ENSDARG00000113512 ddhd1b             CHR_ALT_CTG20_2_1   9787436      \n",
       "43371 ENSDARG00000113512 ddhd1b             CHR_ALT_CTG20_2_1   9787436      \n",
       "43372 ENSDARG00000113512 ddhd1b             CHR_ALT_CTG20_2_1   9787436      \n",
       "      end_position ensembl_peptide_id xtropicalis_homolog_ensembl_gene\n",
       "1     20934666     ENSDARP00000123357 ENSXETG00000026898              \n",
       "2     20934666     ENSDARP00000114467 ENSXETG00000026898              \n",
       "3     14607855     ENSDARP00000140695 ENSXETG00000018133              \n",
       "4     14607855     ENSDARP00000147450 ENSXETG00000018133              \n",
       "5     34683370     ENSDARP00000099496                                 \n",
       "6     24484758     ENSDARP00000052421                                 \n",
       "7     15117508     ENSDARP00000129251 ENSXETG00000012217              \n",
       "8     15117508     ENSDARP00000101105 ENSXETG00000012217              \n",
       "9     38954874     ENSDARP00000012411 ENSXETG00000031075              \n",
       "10    38954874     ENSDARP00000142646 ENSXETG00000031075              \n",
       "11    38954874     ENSDARP00000134652 ENSXETG00000031075              \n",
       "12    39148199     ENSDARP00000129339 ENSXETG00000004743              \n",
       "13    39148199     ENSDARP00000129719 ENSXETG00000004743              \n",
       "14    39148199     ENSDARP00000149720 ENSXETG00000004743              \n",
       "15    39148199     ENSDARP00000155235 ENSXETG00000004743              \n",
       "16    45279411     ENSDARP00000119445 ENSXETG00000006882              \n",
       "17    45279411     ENSDARP00000143739 ENSXETG00000006882              \n",
       "18     8973300     ENSDARP00000123402                                 \n",
       "19     8973300     ENSDARP00000066781                                 \n",
       "20    36356642     ENSDARP00000133585 ENSXETG00000010509              \n",
       "21    36356642     ENSDARP00000133585 ENSXETG00000038303              \n",
       "22    36356642     ENSDARP00000133585 ENSXETG00000001428              \n",
       "23    36356642     ENSDARP00000133585 ENSXETG00000035307              \n",
       "24    36356642     ENSDARP00000133585 ENSXETG00000035638              \n",
       "25    36356642     ENSDARP00000133585 ENSXETG00000036555              \n",
       "26    36356642     ENSDARP00000133585 ENSXETG00000035336              \n",
       "27    36356642     ENSDARP00000133585 ENSXETG00000034481              \n",
       "28    36356642     ENSDARP00000133585 ENSXETG00000035889              \n",
       "29    36356642     ENSDARP00000133585 ENSXETG00000027192              \n",
       "30    28425011     ENSDARP00000122708 ENSXETG00000031018              \n",
       "⋮     ⋮            ⋮                  ⋮                               \n",
       "43343 28960450     ENSDARP00000157662                                 \n",
       "43344 19142350     ENSDARP00000155798                                 \n",
       "43345 31626232     ENSDARP00000135190                                 \n",
       "43346 15021628     ENSDARP00000028581 ENSXETG00000038908              \n",
       "43347  9989947     ENSDARP00000148639 ENSXETG00000038538              \n",
       "43348  9989947     ENSDARP00000154715 ENSXETG00000038538              \n",
       "43349 25396939     ENSDARP00000157436                                 \n",
       "43350    54173     ENSDARP00000135683 ENSXETG00000002497              \n",
       "43351 35108893     ENSDARP00000051296 ENSXETG00000037513              \n",
       "43352 35108893     ENSDARP00000051296 ENSXETG00000016816              \n",
       "43353 42075487     ENSDARP00000139898 ENSXETG00000004038              \n",
       "43354 29308455     ENSDARP00000151471                                 \n",
       "43355 42306597     ENSDARP00000135916 ENSXETG00000017673              \n",
       "43356 42306597     ENSDARP00000149791 ENSXETG00000017673              \n",
       "43357 72261721     ENSDARP00000136405 ENSXETG00000040253              \n",
       "43358 21009442     ENSDARP00000147480                                 \n",
       "43359 15630421     ENSDARP00000109101 ENSXETG00000004467              \n",
       "43360 62061713     ENSDARP00000136664                                 \n",
       "43361 19215380     ENSDARP00000146666                                 \n",
       "43362 19362387     ENSDARP00000135153                                 \n",
       "43363 49481672     ENSDARP00000130801 ENSXETG00000027288              \n",
       "43364 48869632     ENSDARP00000141379                                 \n",
       "43365  3286769     ENSDARP00000132462 ENSXETG00000010703              \n",
       "43366  3286769     ENSDARP00000146047 ENSXETG00000010703              \n",
       "43367  3286769     ENSDARP00000148256 ENSXETG00000010703              \n",
       "43368  3286769     ENSDARP00000130006 ENSXETG00000010703              \n",
       "43369  3286769     ENSDARP00000132912 ENSXETG00000010703              \n",
       "43370  9823395     ENSDARP00000156335                                 \n",
       "43371  9823395     ENSDARP00000145494                                 \n",
       "43372  9823395     ENSDARP00000151727                                 \n",
       "      xtropicalis_homolog_associated_gene_name\n",
       "1     fam162a                                 \n",
       "2     fam162a                                 \n",
       "3     rab9b                                   \n",
       "4     rab9b                                   \n",
       "5                                             \n",
       "6                                             \n",
       "7     samd4a                                  \n",
       "8     samd4a                                  \n",
       "9     creb3l1                                 \n",
       "10    creb3l1                                 \n",
       "11    creb3l1                                 \n",
       "12    cdc42                                   \n",
       "13    cdc42                                   \n",
       "14    cdc42                                   \n",
       "15    cdc42                                   \n",
       "16    adamts13                                \n",
       "17    adamts13                                \n",
       "18                                            \n",
       "19                                            \n",
       "20                                            \n",
       "21                                            \n",
       "22                                            \n",
       "23                                            \n",
       "24                                            \n",
       "25                                            \n",
       "26    ndufaf2                                 \n",
       "27                                            \n",
       "28                                            \n",
       "29    shisa2                                  \n",
       "30    pitpna                                  \n",
       "⋮     ⋮                                       \n",
       "43343                                         \n",
       "43344                                         \n",
       "43345                                         \n",
       "43346 myoc                                    \n",
       "43347 imp3                                    \n",
       "43348 imp3                                    \n",
       "43349                                         \n",
       "43350 stk19                                   \n",
       "43351                                         \n",
       "43352 tmem174.2                               \n",
       "43353 vapa                                    \n",
       "43354                                         \n",
       "43355 ppox                                    \n",
       "43356 ppox                                    \n",
       "43357 rasgrp2                                 \n",
       "43358                                         \n",
       "43359 lpl                                     \n",
       "43360                                         \n",
       "43361                                         \n",
       "43362                                         \n",
       "43363 fcf1                                    \n",
       "43364                                         \n",
       "43365 pak1                                    \n",
       "43366 pak1                                    \n",
       "43367 pak1                                    \n",
       "43368 pak1                                    \n",
       "43369 pak1                                    \n",
       "43370                                         \n",
       "43371                                         \n",
       "43372                                         \n",
       "      xtropicalis_homolog_ensembl_peptide xtropicalis_homolog_chromosome\n",
       "1     ENSXETP00000056418                  KV460586.1                    \n",
       "2     ENSXETP00000056418                  KV460586.1                    \n",
       "3     ENSXETP00000039329                  8                             \n",
       "4     ENSXETP00000039329                  8                             \n",
       "5                                                                       \n",
       "6                                                                       \n",
       "7     ENSXETP00000073845                  8                             \n",
       "8     ENSXETP00000073845                  8                             \n",
       "9     ENSXETP00000062801                  4                             \n",
       "10    ENSXETP00000062801                  4                             \n",
       "11    ENSXETP00000062801                  4                             \n",
       "12    ENSXETP00000030366                  7                             \n",
       "13    ENSXETP00000030366                  7                             \n",
       "14    ENSXETP00000030366                  7                             \n",
       "15    ENSXETP00000030366                  7                             \n",
       "16    ENSXETP00000070357                  KV460557.1                    \n",
       "17    ENSXETP00000070357                  KV460557.1                    \n",
       "18                                                                      \n",
       "19                                                                      \n",
       "20    ENSXETP00000023050                  KV460380.1                    \n",
       "21    ENSXETP00000066651                  KV460380.1                    \n",
       "22    ENSXETP00000003018                  KV460380.1                    \n",
       "23    ENSXETP00000021357                  KV460380.1                    \n",
       "24    ENSXETP00000003015                  KV460380.1                    \n",
       "25    ENSXETP00000077786                  KV460380.1                    \n",
       "26    ENSXETP00000056315                  KV460380.1                    \n",
       "27    ENSXETP00000021354                  KV460380.1                    \n",
       "28    ENSXETP00000066505                  KV460380.1                    \n",
       "29    ENSXETP00000056884                  KV460380.1                    \n",
       "30    ENSXETP00000085667                  2                             \n",
       "⋮     ⋮                                   ⋮                             \n",
       "43343                                                                   \n",
       "43344                                                                   \n",
       "43345                                                                   \n",
       "43346 ENSXETP00000073502                  4                             \n",
       "43347 ENSXETP00000100976                  4                             \n",
       "43348 ENSXETP00000100976                  4                             \n",
       "43349                                                                   \n",
       "43350 ENSXETP00000005317                  8                             \n",
       "43351 ENSXETP00000059943                  1                             \n",
       "43352 ENSXETP00000036664                  1                             \n",
       "43353 ENSXETP00000008738                  6                             \n",
       "43354                                                                   \n",
       "43355 ENSXETP00000038355                  8                             \n",
       "43356 ENSXETP00000038355                  8                             \n",
       "43357 ENSXETP00000094240                  4                             \n",
       "43358                                                                   \n",
       "43359 ENSXETP00000066597                  1                             \n",
       "43360                                                                   \n",
       "43361                                                                   \n",
       "43362                                                                   \n",
       "43363 ENSXETP00000057007                  8                             \n",
       "43364                                                                   \n",
       "43365 ENSXETP00000025828                  2                             \n",
       "43366 ENSXETP00000025828                  2                             \n",
       "43367 ENSXETP00000025828                  2                             \n",
       "43368 ENSXETP00000025828                  2                             \n",
       "43369 ENSXETP00000025828                  2                             \n",
       "43370                                                                   \n",
       "43371                                                                   \n",
       "43372                                                                   \n",
       "      xtropicalis_homolog_chrom_start xtropicalis_homolog_chrom_end\n",
       "1          691                            4121                     \n",
       "2          691                            4121                     \n",
       "3     16957503                        16961130                     \n",
       "4     16957503                        16961130                     \n",
       "5           NA                              NA                     \n",
       "6           NA                              NA                     \n",
       "7     74362645                        74457225                     \n",
       "8     74362645                        74457225                     \n",
       "9      5396261                         5434390                     \n",
       "10     5396261                         5434390                     \n",
       "11     5396261                         5434390                     \n",
       "12    68594070                        68619099                     \n",
       "13    68594070                        68619099                     \n",
       "14    68594070                        68619099                     \n",
       "15    68594070                        68619099                     \n",
       "16       60725                           78242                     \n",
       "17       60725                           78242                     \n",
       "18          NA                              NA                     \n",
       "19          NA                              NA                     \n",
       "20      695815                          755758                     \n",
       "21      585295                          593990                     \n",
       "22      657394                          679990                     \n",
       "23      636297                          637277                     \n",
       "24      629793                          630770                     \n",
       "25      516235                          568510                     \n",
       "26      607003                          613853                     \n",
       "27      585049                          586239                     \n",
       "28      598407                          600057                     \n",
       "29      763981                          765316                     \n",
       "30    39718101                        39751925                     \n",
       "⋮     ⋮                               ⋮                            \n",
       "43343        NA                              NA                    \n",
       "43344        NA                              NA                    \n",
       "43345        NA                              NA                    \n",
       "43346  97460465                        97468737                    \n",
       "43347 107643257                       107708843                    \n",
       "43348 107643257                       107708843                    \n",
       "43349        NA                              NA                    \n",
       "43350  53693914                        53701289                    \n",
       "43351 166687146                       166687814                    \n",
       "43352 166678058                       166678894                    \n",
       "43353  89550547                        89570343                    \n",
       "43354        NA                              NA                    \n",
       "43355 117337070                       117344793                    \n",
       "43356 117337070                       117344793                    \n",
       "43357  43109116                        43140780                    \n",
       "43358        NA                              NA                    \n",
       "43359  77715998                        77736810                    \n",
       "43360        NA                              NA                    \n",
       "43361        NA                              NA                    \n",
       "43362        NA                              NA                    \n",
       "43363  60956283                        60967007                    \n",
       "43364        NA                              NA                    \n",
       "43365 167246839                       167282942                    \n",
       "43366 167246839                       167282942                    \n",
       "43367 167246839                       167282942                    \n",
       "43368 167246839                       167282942                    \n",
       "43369 167246839                       167282942                    \n",
       "43370        NA                              NA                    \n",
       "43371        NA                              NA                    \n",
       "43372        NA                              NA                    \n",
       "      xtropicalis_homolog_subtype xtropicalis_homolog_orthology_type\n",
       "1     Euteleostomi                ortholog_one2one                  \n",
       "2     Euteleostomi                ortholog_one2one                  \n",
       "3     Euteleostomi                ortholog_one2one                  \n",
       "4     Euteleostomi                ortholog_one2one                  \n",
       "5                                                                   \n",
       "6                                                                   \n",
       "7     Gnathostomata               ortholog_one2one                  \n",
       "8     Gnathostomata               ortholog_one2one                  \n",
       "9     Euteleostomi                ortholog_one2one                  \n",
       "10    Euteleostomi                ortholog_one2one                  \n",
       "11    Euteleostomi                ortholog_one2one                  \n",
       "12    Chordata                    ortholog_one2one                  \n",
       "13    Chordata                    ortholog_one2one                  \n",
       "14    Chordata                    ortholog_one2one                  \n",
       "15    Chordata                    ortholog_one2one                  \n",
       "16    Euteleostomi                ortholog_one2one                  \n",
       "17    Euteleostomi                ortholog_one2one                  \n",
       "18                                                                  \n",
       "19                                                                  \n",
       "20    Euteleostomi                ortholog_many2many                \n",
       "21    Euteleostomi                ortholog_many2many                \n",
       "22    Euteleostomi                ortholog_many2many                \n",
       "23    Euteleostomi                ortholog_many2many                \n",
       "24    Euteleostomi                ortholog_many2many                \n",
       "25    Euteleostomi                ortholog_many2many                \n",
       "26    Euteleostomi                ortholog_many2many                \n",
       "27    Euteleostomi                ortholog_many2many                \n",
       "28    Euteleostomi                ortholog_many2many                \n",
       "29    Euteleostomi                ortholog_many2many                \n",
       "30    Euteleostomi                ortholog_one2many                 \n",
       "⋮     ⋮                           ⋮                                 \n",
       "43343                                                               \n",
       "43344                                                               \n",
       "43345                                                               \n",
       "43346 Chordata                    ortholog_one2one                  \n",
       "43347 Euteleostomi                ortholog_one2one                  \n",
       "43348 Euteleostomi                ortholog_one2one                  \n",
       "43349                                                               \n",
       "43350 Gnathostomata               ortholog_one2many                 \n",
       "43351 Euteleostomi                ortholog_one2many                 \n",
       "43352 Euteleostomi                ortholog_one2many                 \n",
       "43353 Gnathostomata               ortholog_one2one                  \n",
       "43354                                                               \n",
       "43355 Euteleostomi                ortholog_one2one                  \n",
       "43356 Euteleostomi                ortholog_one2one                  \n",
       "43357 Euteleostomi                ortholog_one2many                 \n",
       "43358                                                               \n",
       "43359 Euteleostomi                ortholog_one2many                 \n",
       "43360                                                               \n",
       "43361                                                               \n",
       "43362                                                               \n",
       "43363 Vertebrata                  ortholog_one2one                  \n",
       "43364                                                               \n",
       "43365 Euteleostomi                ortholog_one2one                  \n",
       "43366 Euteleostomi                ortholog_one2one                  \n",
       "43367 Euteleostomi                ortholog_one2one                  \n",
       "43368 Euteleostomi                ortholog_one2one                  \n",
       "43369 Euteleostomi                ortholog_one2one                  \n",
       "43370                                                               \n",
       "43371                                                               \n",
       "43372                                                               \n",
       "      xtropicalis_homolog_perc_id xtropicalis_homolog_perc_id_r1\n",
       "1     35.8974                     47.8632                       \n",
       "2     35.8974                     47.8632                       \n",
       "3     75.2475                     75.6219                       \n",
       "4     75.2475                     75.6219                       \n",
       "5          NA                          NA                       \n",
       "6          NA                          NA                       \n",
       "7     70.0581                     67.6966                       \n",
       "8     70.0581                     67.6966                       \n",
       "9     51.9011                     58.4582                       \n",
       "10    51.9011                     58.4582                       \n",
       "11    51.9011                     58.4582                       \n",
       "12    98.4293                     88.6792                       \n",
       "13    98.4293                     88.6792                       \n",
       "14    98.4293                     88.6792                       \n",
       "15    98.4293                     88.6792                       \n",
       "16    38.6515                     42.8889                       \n",
       "17    38.6515                     42.8889                       \n",
       "18         NA                          NA                       \n",
       "19         NA                          NA                       \n",
       "20    39.1437                     39.7516                       \n",
       "21    37.0031                     38.0503                       \n",
       "22    37.0031                     37.6947                       \n",
       "23    39.1437                     39.2638                       \n",
       "24    39.4495                     39.6923                       \n",
       "25    37.9205                     36.9048                       \n",
       "26    37.0031                     36.1194                       \n",
       "27    38.2263                     38.8199                       \n",
       "28    35.4740                     36.8254                       \n",
       "29    32.7217                     33.7539                       \n",
       "30    77.4908                     72.6644                       \n",
       "⋮     ⋮                           ⋮                             \n",
       "43343      NA                          NA                       \n",
       "43344      NA                          NA                       \n",
       "43345      NA                          NA                       \n",
       "43346 46.2025                     43.8878                       \n",
       "43347 87.4317                     87.4317                       \n",
       "43348 87.4317                     87.4317                       \n",
       "43349      NA                          NA                       \n",
       "43350 45.4902                     44.6154                       \n",
       "43351 36.1345                     38.7387                       \n",
       "43352 42.8571                     36.6906                       \n",
       "43353 62.0000                     39.4068                       \n",
       "43354      NA                          NA                       \n",
       "43355 55.3459                     55.3459                       \n",
       "43356 55.3459                     55.3459                       \n",
       "43357 61.3636                     55.7659                       \n",
       "43358      NA                          NA                       \n",
       "43359 56.5558                     59.4650                       \n",
       "43360      NA                          NA                       \n",
       "43361      NA                          NA                       \n",
       "43362      NA                          NA                       \n",
       "43363 89.8990                     90.3553                       \n",
       "43364      NA                          NA                       \n",
       "43365 63.7138                     71.7850                       \n",
       "43366 63.7138                     71.7850                       \n",
       "43367 63.7138                     71.7850                       \n",
       "43368 63.7138                     71.7850                       \n",
       "43369 63.7138                     71.7850                       \n",
       "43370      NA                          NA                       \n",
       "43371      NA                          NA                       \n",
       "43372      NA                          NA                       \n",
       "      xtropicalis_homolog_goc_score xtropicalis_homolog_wga_coverage\n",
       "1     25                             68.86                          \n",
       "2     25                             68.86                          \n",
       "3     25                            100.00                          \n",
       "4     25                            100.00                          \n",
       "5     NA                                NA                          \n",
       "6     NA                                NA                          \n",
       "7     50                             80.53                          \n",
       "8     50                             80.53                          \n",
       "9     75                             60.05                          \n",
       "10    75                             60.05                          \n",
       "11    75                             60.05                          \n",
       "12    75                             94.10                          \n",
       "13    75                             94.10                          \n",
       "14    75                             94.10                          \n",
       "15    75                             94.10                          \n",
       "16     0                             35.79                          \n",
       "17     0                             35.79                          \n",
       "18    NA                                NA                          \n",
       "19    NA                                NA                          \n",
       "20     0                              0.00                          \n",
       "21     0                              0.00                          \n",
       "22     0                             90.99                          \n",
       "23     0                              0.00                          \n",
       "24     0                              0.00                          \n",
       "25     0                             46.38                          \n",
       "26     0                              0.00                          \n",
       "27     0                              0.00                          \n",
       "28     0                             90.59                          \n",
       "29     0                              0.00                          \n",
       "30    25                             92.07                          \n",
       "⋮     ⋮                             ⋮                               \n",
       "43343  NA                              NA                           \n",
       "43344  NA                              NA                           \n",
       "43345  NA                              NA                           \n",
       "43346 100                           65.36                           \n",
       "43347  25                           85.87                           \n",
       "43348  25                           85.87                           \n",
       "43349  NA                              NA                           \n",
       "43350   0                           19.19                           \n",
       "43351 100                           71.58                           \n",
       "43352 100                           46.92                           \n",
       "43353  25                           10.73                           \n",
       "43354  NA                              NA                           \n",
       "43355   0                           66.05                           \n",
       "43356   0                           66.05                           \n",
       "43357   0                           71.28                           \n",
       "43358  NA                              NA                           \n",
       "43359   0                           70.31                           \n",
       "43360  NA                              NA                           \n",
       "43361  NA                              NA                           \n",
       "43362  NA                              NA                           \n",
       "43363  25                           87.83                           \n",
       "43364  NA                              NA                           \n",
       "43365  25                           18.82                           \n",
       "43366  25                           18.82                           \n",
       "43367  25                           18.82                           \n",
       "43368  25                           18.82                           \n",
       "43369  25                           18.82                           \n",
       "43370  NA                              NA                           \n",
       "43371  NA                              NA                           \n",
       "43372  NA                              NA                           \n",
       "      xtropicalis_homolog_orthology_confidence\n",
       "1      1                                      \n",
       "2      1                                      \n",
       "3      1                                      \n",
       "4      1                                      \n",
       "5     NA                                      \n",
       "6     NA                                      \n",
       "7      1                                      \n",
       "8      1                                      \n",
       "9      1                                      \n",
       "10     1                                      \n",
       "11     1                                      \n",
       "12     1                                      \n",
       "13     1                                      \n",
       "14     1                                      \n",
       "15     1                                      \n",
       "16     0                                      \n",
       "17     0                                      \n",
       "18    NA                                      \n",
       "19    NA                                      \n",
       "20     0                                      \n",
       "21     0                                      \n",
       "22     1                                      \n",
       "23     0                                      \n",
       "24     0                                      \n",
       "25     0                                      \n",
       "26     0                                      \n",
       "27     0                                      \n",
       "28     1                                      \n",
       "29     0                                      \n",
       "30     1                                      \n",
       "⋮     ⋮                                       \n",
       "43343 NA                                      \n",
       "43344 NA                                      \n",
       "43345 NA                                      \n",
       "43346  1                                      \n",
       "43347  1                                      \n",
       "43348  1                                      \n",
       "43349 NA                                      \n",
       "43350  0                                      \n",
       "43351  1                                      \n",
       "43352  1                                      \n",
       "43353  0                                      \n",
       "43354 NA                                      \n",
       "43355  1                                      \n",
       "43356  1                                      \n",
       "43357  1                                      \n",
       "43358 NA                                      \n",
       "43359  1                                      \n",
       "43360 NA                                      \n",
       "43361 NA                                      \n",
       "43362 NA                                      \n",
       "43363  1                                      \n",
       "43364 NA                                      \n",
       "43365  0                                      \n",
       "43366  0                                      \n",
       "43367  0                                      \n",
       "43368  0                                      \n",
       "43369  0                                      \n",
       "43370 NA                                      \n",
       "43371 NA                                      \n",
       "43372 NA                                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "homology_tbl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a70aaaa4-683c-4b5a-a4db-c56f22169077",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "seurat",
   "language": "R",
   "name": "seurat"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
