{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "5436d80a-c36e-4e38-89c6-6bc2322166e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from biomart import BiomartServer\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "import os.path\n",
    "import os\n",
    "import itertools as it\n",
    "import numpy as np\n",
    "import plotly.express as px\n",
    "from plotly.graph_objects import Figure\n",
    "\n",
    "### 1 TPM\n",
    "\n",
    "species_name_dict = {\n",
    "    \"drerio\": \"zebrafish_gene_name_ct_final.csv\",\n",
    "    \"xtropicalis\": \"xenopus_gene_name_ct_final.csv\"\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d732bd9b-771e-4041-b9fd-92e777130744",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_cross_species_spec_category_heatmap(mapping_both, species_1, species_2, group_enhanced=False) -> Figure:\n",
    "    if not group_enhanced:\n",
    "        mapping_both.loc[\n",
    "            mapping_both[f\"spec_category_{species_1}\"] == 'group enhanced', [f\"spec_category_{species_1}\"]] = 'enhanced'\n",
    "        mapping_both.loc[mapping_both[f\"spec_category_{species_1}\"] == 'cell type enhanced', [\n",
    "            f\"spec_category_{species_1}\"]] = 'enhanced'\n",
    "\n",
    "        mapping_both.loc[\n",
    "            mapping_both[f\"spec_category_{species_2}\"] == 'group enhanced', [f\"spec_category_{species_2}\"]] = 'enhanced'\n",
    "        mapping_both.loc[mapping_both[f\"spec_category_{species_2}\"] == 'cell type enhanced', [\n",
    "            f\"spec_category_{species_2}\"]] = 'enhanced'\n",
    "\n",
    "    df = mapping_both.loc[mapping_both[f\"{species_2}_homolog_orthology_type\"] == 'ortholog_one2one'].groupby(\n",
    "        [f\"spec_category_{species_1}\"])[f\"spec_category_{species_2}\"] \\\n",
    "        .value_counts().to_frame('count').reset_index()\n",
    "\n",
    "    species1_counts = df.groupby(f\"spec_category_{species_1}\")['count'].sum()\n",
    "    species2_counts = df.groupby(f\"spec_category_{species_2}\")['count'].sum()\n",
    "\n",
    "    df['Species1_Percentage'] = df.apply(\n",
    "        lambda row:\n",
    "        (row['count'] / species1_counts[row[f\"spec_category_{species_1}\"]]) * 100,\n",
    "        axis=1)\n",
    "    df['Species2_Percentage'] = df.apply(\n",
    "        lambda row:\n",
    "        (row['count'] / species2_counts[row[f\"spec_category_{species_2}\"]]) * 100,\n",
    "        axis=1)\n",
    "    df['Harmonic_Mean'] = 2 / ((1 / df['Species1_Percentage']) + (1 / df['Species2_Percentage']))\n",
    "\n",
    "    heatmap_data = df.pivot(columns=f\"spec_category_{species_1}\",\n",
    "                            index=f\"spec_category_{species_2}\",\n",
    "                            values='Harmonic_Mean')\n",
    "\n",
    "    heatmap_data = np.round(heatmap_data.astype(\"float32\"), 2)\n",
    "\n",
    "    if not group_enhanced:\n",
    "\n",
    "        order = ['cell type enriched',\n",
    "                 'group enriched',\n",
    "                 'enhanced',\n",
    "                 'low cell type specificity',\n",
    "                 'lowly expressed']\n",
    "    else:\n",
    "        order = ['cell type enriched',\n",
    "                 'group enriched',\n",
    "                 'cell type enhanced',\n",
    "                 'group enhanced',\n",
    "                 'low cell type specificity',\n",
    "                 'lowly expressed']\n",
    "\n",
    "    fig = px.imshow(heatmap_data[order].reindex(order), title='One2one orthologs',\n",
    "                    text_auto=True,\n",
    "                    width=750, height=600,\n",
    "                    labels=dict(x=species_1, y=species_2, color=\"% overlap\"))\n",
    "    fig['layout']['yaxis']['autorange'] = \"reversed\"\n",
    "    fig.update_yaxes(autorange=True)\n",
    "\n",
    "    return heatmap_data, fig\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9c562c4-e541-49c6-b21d-d231a10e4351",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "d83ab28d-b0e1-4b39-91ff-e35de3103740",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "start drerio, xtropicalis\n",
      "total genes from drerio: 17330\n",
      "total genes from drerio: 17330\n",
      "ensembl homology mapped genes from drerio: 7717\n",
      "total genes from xtropicalis: 9661\n",
      "ensembl homology mapped genes from xtropicalis: 6274\n",
      "finished drerio, xtropicalis\n"
     ]
    }
   ],
   "source": [
    "\n",
    "species_1, species_2 = (\"drerio\", \"xtropicalis\")\n",
    "\n",
    "if not os.path.isfile(f\"o2o_heatmap_figs/{species_1}_{species_2}_o2o_heatmap_1TPM.pdf\"):\n",
    "    print(f\"start {species_1}, {species_2}\")\n",
    " \n",
    "    classes_sp1 = pd.read_csv(species_name_dict.get(species_1))\n",
    "    classes_sp2 = pd.read_csv(species_name_dict.get(species_2))\n",
    "    \n",
    "    server = BiomartServer(\"http://www.ensembl.org/biomart\")\n",
    "    # Select the Ensembl Gene ID dataset\n",
    "    ensembl = server.datasets[f\"{species_1}_gene_ensembl\"]\n",
    "    \n",
    "    # Define the columns for the query\n",
    "    columns = ['ensembl_gene_id', 'external_gene_name', f\"{species_2}_homolog_ensembl_gene\",\n",
    "               f\"{species_2}_homolog_orthology_type\", f\"{species_2}_homolog_associated_gene_name\"]\n",
    "    \n",
    "    # Query BioMart for the mappings of orthologs genes\n",
    "    print(f\"total genes from {species_1}: {len(set(classes_sp1.gene))}\")\n",
    "    \n",
    "    response = ensembl.search({\n",
    "        'query': classes_sp1.gene.tolist(),\n",
    "        'attributes': columns,\n",
    "        'mart_instance': 'ensembl'\n",
    "    })\n",
    "    \n",
    "\n",
    "    mapping_df = pd.read_csv(response.url, sep='\\t', header=None, names=columns)\n",
    "    \n",
    "    mapping_df = mapping_df.loc[mapping_df[f\"{species_2}_homolog_associated_gene_name\"].notnull()]\n",
    "    \n",
    "    mapping_df = mapping_df.loc[mapping_df[f\"{species_2}_homolog_orthology_type\"]== 'ortholog_one2one', :]\n",
    "    \n",
    "    print(f\"total genes from {species_1}: {len(set(classes_sp1.gene))}\")\n",
    "    unique_genes_mapped_sp1 = len(list(set(classes_sp1.gene) & set(mapping_df.external_gene_name)))\n",
    "    \n",
    "    print(f\"ensembl homology mapped genes from {species_1}: {unique_genes_mapped_sp1}\")\n",
    "    \n",
    "    mapping_sp1 = pd.merge(mapping_df, classes_sp1, left_on='external_gene_name', right_on='gene', how='inner')\n",
    "    unique_genes_mapped_sp2 = len(list(set(classes_sp2.gene) & set(mapping_df[f\"{species_2}_homolog_associated_gene_name\"])))\n",
    "        \n",
    "    print(f\"total genes from {species_2}: {len(set(classes_sp2.gene))}\")\n",
    "    print(f\"ensembl homology mapped genes from {species_2}: {unique_genes_mapped_sp2}\")\n",
    "    \n",
    "    mapping_both = pd.merge(mapping_sp1, classes_sp2, left_on=f\"{species_2}_homolog_associated_gene_name\", right_on='gene', how='inner',\n",
    "                    suffixes=(f\"_{species_1}\", f\"_{species_2}\")).drop(columns=['external_gene_name'])\n",
    "\n",
    "    mapping_both.to_csv(f\"./homology_mapped_o2o_{species_1}_{species_2}_1TPM.csv\")\n",
    "\n",
    "    \n",
    "    heatmap_data, o2o_heatmap = plot_cross_species_spec_category_heatmap(species_1=species_1, \n",
    "                                                                         species_2=species_2, \n",
    "                                                                         mapping_both=mapping_both, \n",
    "                                                                         group_enhanced=True)\n",
    "    \n",
    "    o2o_heatmap.write_image(f\"./{species_1}_{species_2}_o2o_heatmap_1TPM.pdf\")\n",
    "    \n",
    "    heatmap_data.to_csv(f\"./o2o_heatmap_data_{species_1}_{species_2}_1TPM.csv\")\n",
    "    \n",
    "    print(f\"finished {species_1}, {species_2}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a2c0e999-66b3-4f28-975d-e9c2c136931f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>xtropicalis_homolog_ensembl_gene</th>\n",
       "      <th>xtropicalis_homolog_orthology_type</th>\n",
       "      <th>xtropicalis_homolog_associated_gene_name</th>\n",
       "      <th>gene_drerio</th>\n",
       "      <th>mean_exp_drerio</th>\n",
       "      <th>min_exp_drerio</th>\n",
       "      <th>max_exp_drerio</th>\n",
       "      <th>max_2nd_drerio</th>\n",
       "      <th>n_exp_drerio</th>\n",
       "      <th>...</th>\n",
       "      <th>enrichment_group_xtropicalis</th>\n",
       "      <th>n_enriched_xtropicalis</th>\n",
       "      <th>max_2nd_or_lim_xtropicalis</th>\n",
       "      <th>exps_enhanced_xtropicalis</th>\n",
       "      <th>n_enhanced_xtropicalis</th>\n",
       "      <th>enhanced_in_xtropicalis</th>\n",
       "      <th>spec_category_xtropicalis</th>\n",
       "      <th>dist_category_xtropicalis</th>\n",
       "      <th>spec_score_xtropicalis</th>\n",
       "      <th>enriched_groups_xtropicalis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSDARG00000069301</td>\n",
       "      <td>ENSXETG00000042724</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>tmem177</td>\n",
       "      <td>tmem177</td>\n",
       "      <td>0.183647</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6351</td>\n",
       "      <td>0.4179</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.3802</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSDARG00000104901</td>\n",
       "      <td>ENSXETG00000021099</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>ostc</td>\n",
       "      <td>ostc</td>\n",
       "      <td>2.886530</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.9463</td>\n",
       "      <td>5.4093</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>Cement_gland_primordium;Endothelial;Epidermal_...</td>\n",
       "      <td>0</td>\n",
       "      <td>19.8031</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 90%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSDARG00000103391</td>\n",
       "      <td>ENSXETG00000018133</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>rab9b</td>\n",
       "      <td>rab9b</td>\n",
       "      <td>0.040903</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1370</td>\n",
       "      <td>0.1298</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.3231</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSDARG00000030463</td>\n",
       "      <td>ENSXETG00000047601</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>tppp3</td>\n",
       "      <td>tppp3</td>\n",
       "      <td>0.199995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.2683</td>\n",
       "      <td>0.9083</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1266</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSDARG00000070447</td>\n",
       "      <td>ENSXETG00000017909</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>slc39a9</td>\n",
       "      <td>slc39a9</td>\n",
       "      <td>0.092472</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6388</td>\n",
       "      <td>0.1795</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Blastula;Cement_gland_primordium;Endoderm;Germ...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.2668</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 30%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5632</th>\n",
       "      <td>ENSDARG00000077677</td>\n",
       "      <td>ENSXETG00000010500</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>pdgfd</td>\n",
       "      <td>pdgfd</td>\n",
       "      <td>0.011022</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0649</td>\n",
       "      <td>0.0499</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1142</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5633</th>\n",
       "      <td>ENSDARG00000102333</td>\n",
       "      <td>ENSXETG00000027288</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>fcf1</td>\n",
       "      <td>fcf1</td>\n",
       "      <td>1.060507</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.1555</td>\n",
       "      <td>2.1328</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>Blastula;Blood;Endoderm;Endothelial;Epidermal_...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.3266</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 30%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5634</th>\n",
       "      <td>ENSDARG00000005713</td>\n",
       "      <td>ENSXETG00000025507</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>ethe1</td>\n",
       "      <td>ethe1</td>\n",
       "      <td>0.088758</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6533</td>\n",
       "      <td>0.2048</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Myeloid_progenitors</td>\n",
       "      <td>1</td>\n",
       "      <td>4.3715</td>\n",
       "      <td>[17.486000061035156]</td>\n",
       "      <td>1</td>\n",
       "      <td>Myeloid_progenitors</td>\n",
       "      <td>cell type enriched</td>\n",
       "      <td>expressed in less than 30%</td>\n",
       "      <td>6.348849</td>\n",
       "      <td>Myeloid_progenitors</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5635</th>\n",
       "      <td>ENSDARG00000008623</td>\n",
       "      <td>ENSXETG00000005742</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>cops3</td>\n",
       "      <td>cops3</td>\n",
       "      <td>0.569348</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.3495</td>\n",
       "      <td>0.9682</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Blastula;Blood;Cement_gland_primordium;Endoder...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.5441</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 30%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5636</th>\n",
       "      <td>ENSDARG00000103959</td>\n",
       "      <td>ENSXETG00000010703</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>pak1</td>\n",
       "      <td>pak1</td>\n",
       "      <td>0.466890</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.4268</td>\n",
       "      <td>1.0377</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>Notochord</td>\n",
       "      <td>1</td>\n",
       "      <td>0.5438</td>\n",
       "      <td>[1.7976000308990479]</td>\n",
       "      <td>1</td>\n",
       "      <td>Notochord</td>\n",
       "      <td>cell type enhanced</td>\n",
       "      <td>expressed in single</td>\n",
       "      <td>8.317837</td>\n",
       "      <td>Notochord</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5637 rows × 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ensembl_gene_id xtropicalis_homolog_ensembl_gene  \\\n",
       "0     ENSDARG00000069301               ENSXETG00000042724   \n",
       "1     ENSDARG00000104901               ENSXETG00000021099   \n",
       "2     ENSDARG00000103391               ENSXETG00000018133   \n",
       "3     ENSDARG00000030463               ENSXETG00000047601   \n",
       "4     ENSDARG00000070447               ENSXETG00000017909   \n",
       "...                  ...                              ...   \n",
       "5632  ENSDARG00000077677               ENSXETG00000010500   \n",
       "5633  ENSDARG00000102333               ENSXETG00000027288   \n",
       "5634  ENSDARG00000005713               ENSXETG00000025507   \n",
       "5635  ENSDARG00000008623               ENSXETG00000005742   \n",
       "5636  ENSDARG00000103959               ENSXETG00000010703   \n",
       "\n",
       "     xtropicalis_homolog_orthology_type  \\\n",
       "0                      ortholog_one2one   \n",
       "1                      ortholog_one2one   \n",
       "2                      ortholog_one2one   \n",
       "3                      ortholog_one2one   \n",
       "4                      ortholog_one2one   \n",
       "...                                 ...   \n",
       "5632                   ortholog_one2one   \n",
       "5633                   ortholog_one2one   \n",
       "5634                   ortholog_one2one   \n",
       "5635                   ortholog_one2one   \n",
       "5636                   ortholog_one2one   \n",
       "\n",
       "     xtropicalis_homolog_associated_gene_name gene_drerio  mean_exp_drerio  \\\n",
       "0                                     tmem177     tmem177         0.183647   \n",
       "1                                        ostc        ostc         2.886530   \n",
       "2                                       rab9b       rab9b         0.040903   \n",
       "3                                       tppp3       tppp3         0.199995   \n",
       "4                                     slc39a9     slc39a9         0.092472   \n",
       "...                                       ...         ...              ...   \n",
       "5632                                    pdgfd       pdgfd         0.011022   \n",
       "5633                                     fcf1        fcf1         1.060507   \n",
       "5634                                    ethe1       ethe1         0.088758   \n",
       "5635                                    cops3       cops3         0.569348   \n",
       "5636                                     pak1        pak1         0.466890   \n",
       "\n",
       "      min_exp_drerio  max_exp_drerio  max_2nd_drerio  n_exp_drerio  ...  \\\n",
       "0                0.0          0.6351          0.4179             0  ...   \n",
       "1                0.0         12.9463          5.4093            39  ...   \n",
       "2                0.0          0.1370          0.1298             0  ...   \n",
       "3                0.0          6.2683          0.9083             1  ...   \n",
       "4                0.0          0.6388          0.1795             0  ...   \n",
       "...              ...             ...             ...           ...  ...   \n",
       "5632             0.0          0.0649          0.0499             0  ...   \n",
       "5633             0.0          2.1555          2.1328            18  ...   \n",
       "5634             0.0          0.6533          0.2048             0  ...   \n",
       "5635             0.0          1.3495          0.9682             1  ...   \n",
       "5636             0.0          5.4268          1.0377             2  ...   \n",
       "\n",
       "                           enrichment_group_xtropicalis  \\\n",
       "0                                                   NaN   \n",
       "1     Cement_gland_primordium;Endothelial;Epidermal_...   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4     Blastula;Cement_gland_primordium;Endoderm;Germ...   \n",
       "...                                                 ...   \n",
       "5632                                                NaN   \n",
       "5633  Blastula;Blood;Endoderm;Endothelial;Epidermal_...   \n",
       "5634                                Myeloid_progenitors   \n",
       "5635  Blastula;Blood;Cement_gland_primordium;Endoder...   \n",
       "5636                                          Notochord   \n",
       "\n",
       "     n_enriched_xtropicalis max_2nd_or_lim_xtropicalis  \\\n",
       "0                         0                     0.3802   \n",
       "1                         0                    19.8031   \n",
       "2                         0                     0.3231   \n",
       "3                         0                     0.1266   \n",
       "4                         0                     2.2668   \n",
       "...                     ...                        ...   \n",
       "5632                      0                     0.1142   \n",
       "5633                      0                     2.3266   \n",
       "5634                      1                     4.3715   \n",
       "5635                      0                     2.5441   \n",
       "5636                      1                     0.5438   \n",
       "\n",
       "      exps_enhanced_xtropicalis n_enhanced_xtropicalis  \\\n",
       "0                            []                      0   \n",
       "1                            []                      0   \n",
       "2                            []                      0   \n",
       "3                            []                      0   \n",
       "4                            []                      0   \n",
       "...                         ...                    ...   \n",
       "5632                         []                      0   \n",
       "5633                         []                      0   \n",
       "5634       [17.486000061035156]                      1   \n",
       "5635                         []                      0   \n",
       "5636       [1.7976000308990479]                      1   \n",
       "\n",
       "     enhanced_in_xtropicalis  spec_category_xtropicalis  \\\n",
       "0                        NaN            lowly expressed   \n",
       "1                        NaN  low cell type specificity   \n",
       "2                        NaN            lowly expressed   \n",
       "3                        NaN            lowly expressed   \n",
       "4                        NaN  low cell type specificity   \n",
       "...                      ...                        ...   \n",
       "5632                     NaN            lowly expressed   \n",
       "5633                     NaN  low cell type specificity   \n",
       "5634     Myeloid_progenitors         cell type enriched   \n",
       "5635                     NaN  low cell type specificity   \n",
       "5636               Notochord         cell type enhanced   \n",
       "\n",
       "       dist_category_xtropicalis  spec_score_xtropicalis  \\\n",
       "0                lowly expressed                0.000000   \n",
       "1          expressed in over 90%                0.000000   \n",
       "2                lowly expressed                0.000000   \n",
       "3                lowly expressed                0.000000   \n",
       "4          expressed in over 30%                0.000000   \n",
       "...                          ...                     ...   \n",
       "5632             lowly expressed                0.000000   \n",
       "5633       expressed in over 30%                0.000000   \n",
       "5634  expressed in less than 30%                6.348849   \n",
       "5635       expressed in over 30%                0.000000   \n",
       "5636         expressed in single                8.317837   \n",
       "\n",
       "      enriched_groups_xtropicalis  \n",
       "0                             NaN  \n",
       "1                             NaN  \n",
       "2                             NaN  \n",
       "3                             NaN  \n",
       "4                             NaN  \n",
       "...                           ...  \n",
       "5632                          NaN  \n",
       "5633                          NaN  \n",
       "5634          Myeloid_progenitors  \n",
       "5635                          NaN  \n",
       "5636                    Notochord  \n",
       "\n",
       "[5637 rows x 56 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping_both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "f903e67b-222d-45d3-b755-2ef9d3bfc701",
   "metadata": {},
   "outputs": [],
   "source": [
    "import plotly.io as pio\n",
    "pio.renderers.default='iframe'\n",
    "from plotly.graph_objects import Figure\n",
    "import plotly.express as px"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "9f61c015-aaae-4d68-be47-a4e5f580cf5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_categories_pie(data, kind='spec_category', title=None) -> Figure:\n",
    "    \"\"\"\n",
    "\n",
    "    :param data: the categories output by hpa_gene_classification\n",
    "    :param kind: the kind of category to plot, choose between spec_category (default) and dist_category\n",
    "    :param title: title of the plot, usually the species and dataset name\n",
    "    :return: a plotly.graph_objects.Figure of Pie plot\n",
    "    \"\"\"\n",
    "    if kind not in ('spec_category', 'dist_category'):\n",
    "        raise ValueError('kind should be either spec_category or dist_category')\n",
    "\n",
    "    counts = data[kind].value_counts().to_frame().reset_index().rename(\n",
    "        columns={kind: \"counts\", \"index\": 'gene_category'})\n",
    "    fig = px.pie(counts, values='counts', names='gene_category', title=title)\n",
    "    fig.show(renderer='iframe')\n",
    "\n",
    "    return fig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "594bc5d1-71bf-46c8-a953-98c43140955e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_38.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_38.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_categories_pie(classes_sp1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "77ab1032-425f-4c2c-8d63-fc55fceb8f8f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_39.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_39.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_categories_pie(classes_sp1.loc[classes_sp1.gene.isin(mapping_both.gene_drerio),])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "4e2bff7b-a176-4406-a867-436e71341b39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_40.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_40.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_categories_pie(classes_sp2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8947c4b2-68f6-427d-8cb9-294c1f20c2ee",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "cad9eccb-3c81-4d98-97ce-39cc189645bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>xtropicalis_homolog_ensembl_gene</th>\n",
       "      <th>xtropicalis_homolog_orthology_type</th>\n",
       "      <th>xtropicalis_homolog_associated_gene_name</th>\n",
       "      <th>gene_drerio</th>\n",
       "      <th>mean_exp_drerio</th>\n",
       "      <th>min_exp_drerio</th>\n",
       "      <th>max_exp_drerio</th>\n",
       "      <th>max_2nd_drerio</th>\n",
       "      <th>n_exp_drerio</th>\n",
       "      <th>...</th>\n",
       "      <th>enrichment_group_xtropicalis</th>\n",
       "      <th>n_enriched_xtropicalis</th>\n",
       "      <th>max_2nd_or_lim_xtropicalis</th>\n",
       "      <th>exps_enhanced_xtropicalis</th>\n",
       "      <th>n_enhanced_xtropicalis</th>\n",
       "      <th>enhanced_in_xtropicalis</th>\n",
       "      <th>spec_category_xtropicalis</th>\n",
       "      <th>dist_category_xtropicalis</th>\n",
       "      <th>spec_score_xtropicalis</th>\n",
       "      <th>enriched_groups_xtropicalis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSDARG00000069301</td>\n",
       "      <td>ENSXETG00000042724</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>tmem177</td>\n",
       "      <td>tmem177</td>\n",
       "      <td>0.183647</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6351</td>\n",
       "      <td>0.4179</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.3802</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSDARG00000104901</td>\n",
       "      <td>ENSXETG00000021099</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>ostc</td>\n",
       "      <td>ostc</td>\n",
       "      <td>2.886530</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.9463</td>\n",
       "      <td>5.4093</td>\n",
       "      <td>39</td>\n",
       "      <td>...</td>\n",
       "      <td>Cement_gland_primordium;Endothelial;Epidermal_...</td>\n",
       "      <td>0</td>\n",
       "      <td>19.8031</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 90%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSDARG00000103391</td>\n",
       "      <td>ENSXETG00000018133</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>rab9b</td>\n",
       "      <td>rab9b</td>\n",
       "      <td>0.040903</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1370</td>\n",
       "      <td>0.1298</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.3231</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSDARG00000030463</td>\n",
       "      <td>ENSXETG00000047601</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>tppp3</td>\n",
       "      <td>tppp3</td>\n",
       "      <td>0.199995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.2683</td>\n",
       "      <td>0.9083</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1266</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSDARG00000070447</td>\n",
       "      <td>ENSXETG00000017909</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>slc39a9</td>\n",
       "      <td>slc39a9</td>\n",
       "      <td>0.092472</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6388</td>\n",
       "      <td>0.1795</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Blastula;Cement_gland_primordium;Endoderm;Germ...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.2668</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 30%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5632</th>\n",
       "      <td>ENSDARG00000077677</td>\n",
       "      <td>ENSXETG00000010500</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>pdgfd</td>\n",
       "      <td>pdgfd</td>\n",
       "      <td>0.011022</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0649</td>\n",
       "      <td>0.0499</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.1142</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>lowly expressed</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5633</th>\n",
       "      <td>ENSDARG00000102333</td>\n",
       "      <td>ENSXETG00000027288</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>fcf1</td>\n",
       "      <td>fcf1</td>\n",
       "      <td>1.060507</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.1555</td>\n",
       "      <td>2.1328</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>Blastula;Blood;Endoderm;Endothelial;Epidermal_...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.3266</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 30%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5634</th>\n",
       "      <td>ENSDARG00000005713</td>\n",
       "      <td>ENSXETG00000025507</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>ethe1</td>\n",
       "      <td>ethe1</td>\n",
       "      <td>0.088758</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6533</td>\n",
       "      <td>0.2048</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Myeloid_progenitors</td>\n",
       "      <td>1</td>\n",
       "      <td>4.3715</td>\n",
       "      <td>[17.486000061035156]</td>\n",
       "      <td>1</td>\n",
       "      <td>Myeloid_progenitors</td>\n",
       "      <td>cell type enriched</td>\n",
       "      <td>expressed in less than 30%</td>\n",
       "      <td>6.348849</td>\n",
       "      <td>Myeloid_progenitors</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5635</th>\n",
       "      <td>ENSDARG00000008623</td>\n",
       "      <td>ENSXETG00000005742</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>cops3</td>\n",
       "      <td>cops3</td>\n",
       "      <td>0.569348</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.3495</td>\n",
       "      <td>0.9682</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Blastula;Blood;Cement_gland_primordium;Endoder...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.5441</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low cell type specificity</td>\n",
       "      <td>expressed in over 30%</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5636</th>\n",
       "      <td>ENSDARG00000103959</td>\n",
       "      <td>ENSXETG00000010703</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>pak1</td>\n",
       "      <td>pak1</td>\n",
       "      <td>0.466890</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.4268</td>\n",
       "      <td>1.0377</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>Notochord</td>\n",
       "      <td>1</td>\n",
       "      <td>0.5438</td>\n",
       "      <td>[1.7976000308990479]</td>\n",
       "      <td>1</td>\n",
       "      <td>Notochord</td>\n",
       "      <td>cell type enhanced</td>\n",
       "      <td>expressed in single</td>\n",
       "      <td>8.317837</td>\n",
       "      <td>Notochord</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5637 rows × 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ensembl_gene_id xtropicalis_homolog_ensembl_gene  \\\n",
       "0     ENSDARG00000069301               ENSXETG00000042724   \n",
       "1     ENSDARG00000104901               ENSXETG00000021099   \n",
       "2     ENSDARG00000103391               ENSXETG00000018133   \n",
       "3     ENSDARG00000030463               ENSXETG00000047601   \n",
       "4     ENSDARG00000070447               ENSXETG00000017909   \n",
       "...                  ...                              ...   \n",
       "5632  ENSDARG00000077677               ENSXETG00000010500   \n",
       "5633  ENSDARG00000102333               ENSXETG00000027288   \n",
       "5634  ENSDARG00000005713               ENSXETG00000025507   \n",
       "5635  ENSDARG00000008623               ENSXETG00000005742   \n",
       "5636  ENSDARG00000103959               ENSXETG00000010703   \n",
       "\n",
       "     xtropicalis_homolog_orthology_type  \\\n",
       "0                      ortholog_one2one   \n",
       "1                      ortholog_one2one   \n",
       "2                      ortholog_one2one   \n",
       "3                      ortholog_one2one   \n",
       "4                      ortholog_one2one   \n",
       "...                                 ...   \n",
       "5632                   ortholog_one2one   \n",
       "5633                   ortholog_one2one   \n",
       "5634                   ortholog_one2one   \n",
       "5635                   ortholog_one2one   \n",
       "5636                   ortholog_one2one   \n",
       "\n",
       "     xtropicalis_homolog_associated_gene_name gene_drerio  mean_exp_drerio  \\\n",
       "0                                     tmem177     tmem177         0.183647   \n",
       "1                                        ostc        ostc         2.886530   \n",
       "2                                       rab9b       rab9b         0.040903   \n",
       "3                                       tppp3       tppp3         0.199995   \n",
       "4                                     slc39a9     slc39a9         0.092472   \n",
       "...                                       ...         ...              ...   \n",
       "5632                                    pdgfd       pdgfd         0.011022   \n",
       "5633                                     fcf1        fcf1         1.060507   \n",
       "5634                                    ethe1       ethe1         0.088758   \n",
       "5635                                    cops3       cops3         0.569348   \n",
       "5636                                     pak1        pak1         0.466890   \n",
       "\n",
       "      min_exp_drerio  max_exp_drerio  max_2nd_drerio  n_exp_drerio  ...  \\\n",
       "0                0.0          0.6351          0.4179             0  ...   \n",
       "1                0.0         12.9463          5.4093            39  ...   \n",
       "2                0.0          0.1370          0.1298             0  ...   \n",
       "3                0.0          6.2683          0.9083             1  ...   \n",
       "4                0.0          0.6388          0.1795             0  ...   \n",
       "...              ...             ...             ...           ...  ...   \n",
       "5632             0.0          0.0649          0.0499             0  ...   \n",
       "5633             0.0          2.1555          2.1328            18  ...   \n",
       "5634             0.0          0.6533          0.2048             0  ...   \n",
       "5635             0.0          1.3495          0.9682             1  ...   \n",
       "5636             0.0          5.4268          1.0377             2  ...   \n",
       "\n",
       "                           enrichment_group_xtropicalis  \\\n",
       "0                                                   NaN   \n",
       "1     Cement_gland_primordium;Endothelial;Epidermal_...   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4     Blastula;Cement_gland_primordium;Endoderm;Germ...   \n",
       "...                                                 ...   \n",
       "5632                                                NaN   \n",
       "5633  Blastula;Blood;Endoderm;Endothelial;Epidermal_...   \n",
       "5634                                Myeloid_progenitors   \n",
       "5635  Blastula;Blood;Cement_gland_primordium;Endoder...   \n",
       "5636                                          Notochord   \n",
       "\n",
       "     n_enriched_xtropicalis max_2nd_or_lim_xtropicalis  \\\n",
       "0                         0                     0.3802   \n",
       "1                         0                    19.8031   \n",
       "2                         0                     0.3231   \n",
       "3                         0                     0.1266   \n",
       "4                         0                     2.2668   \n",
       "...                     ...                        ...   \n",
       "5632                      0                     0.1142   \n",
       "5633                      0                     2.3266   \n",
       "5634                      1                     4.3715   \n",
       "5635                      0                     2.5441   \n",
       "5636                      1                     0.5438   \n",
       "\n",
       "      exps_enhanced_xtropicalis n_enhanced_xtropicalis  \\\n",
       "0                            []                      0   \n",
       "1                            []                      0   \n",
       "2                            []                      0   \n",
       "3                            []                      0   \n",
       "4                            []                      0   \n",
       "...                         ...                    ...   \n",
       "5632                         []                      0   \n",
       "5633                         []                      0   \n",
       "5634       [17.486000061035156]                      1   \n",
       "5635                         []                      0   \n",
       "5636       [1.7976000308990479]                      1   \n",
       "\n",
       "     enhanced_in_xtropicalis  spec_category_xtropicalis  \\\n",
       "0                        NaN            lowly expressed   \n",
       "1                        NaN  low cell type specificity   \n",
       "2                        NaN            lowly expressed   \n",
       "3                        NaN            lowly expressed   \n",
       "4                        NaN  low cell type specificity   \n",
       "...                      ...                        ...   \n",
       "5632                     NaN            lowly expressed   \n",
       "5633                     NaN  low cell type specificity   \n",
       "5634     Myeloid_progenitors         cell type enriched   \n",
       "5635                     NaN  low cell type specificity   \n",
       "5636               Notochord         cell type enhanced   \n",
       "\n",
       "       dist_category_xtropicalis  spec_score_xtropicalis  \\\n",
       "0                lowly expressed                0.000000   \n",
       "1          expressed in over 90%                0.000000   \n",
       "2                lowly expressed                0.000000   \n",
       "3                lowly expressed                0.000000   \n",
       "4          expressed in over 30%                0.000000   \n",
       "...                          ...                     ...   \n",
       "5632             lowly expressed                0.000000   \n",
       "5633       expressed in over 30%                0.000000   \n",
       "5634  expressed in less than 30%                6.348849   \n",
       "5635       expressed in over 30%                0.000000   \n",
       "5636         expressed in single                8.317837   \n",
       "\n",
       "      enriched_groups_xtropicalis  \n",
       "0                             NaN  \n",
       "1                             NaN  \n",
       "2                             NaN  \n",
       "3                             NaN  \n",
       "4                             NaN  \n",
       "...                           ...  \n",
       "5632                          NaN  \n",
       "5633                          NaN  \n",
       "5634          Myeloid_progenitors  \n",
       "5635                          NaN  \n",
       "5636                    Notochord  \n",
       "\n",
       "[5637 rows x 56 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping_both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "0e8d87bf-c987-4595-aa5a-08f60d7dd456",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_42.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<iframe\n",
       "    scrolling=\"no\"\n",
       "    width=\"100%\"\n",
       "    height=\"545px\"\n",
       "    src=\"iframe_figures/figure_42.html\"\n",
       "    frameborder=\"0\"\n",
       "    allowfullscreen\n",
       "></iframe>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_categories_pie(classes_sp2.loc[classes_sp2.gene.isin(mapping_both.xtropicalis_homolog_associated_gene_name),])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "9075d26d-d785-4161-862b-474ebce599b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_cross_species_dist_category_heatmap(mapping_both, species_1, species_2) -> Figure:\n",
    "\n",
    "    df = mapping_both.loc[mapping_both[f\"{species_2}_homolog_orthology_type\"] == 'ortholog_one2one'].groupby(\n",
    "        [f\"dist_category_{species_1}\"])[f\"dist_category_{species_2}\"] \\\n",
    "        .value_counts().to_frame('count').reset_index()\n",
    "\n",
    "    species1_counts = df.groupby(f\"dist_category_{species_1}\")['count'].sum()\n",
    "    species2_counts = df.groupby(f\"dist_category_{species_2}\")['count'].sum()\n",
    "\n",
    "    df['Species1_Percentage'] = df.apply(\n",
    "        lambda row:\n",
    "        (row['count'] / species1_counts[row[f\"dist_category_{species_1}\"]]) * 100,\n",
    "        axis=1)\n",
    "    df['Species2_Percentage'] = df.apply(\n",
    "        lambda row:\n",
    "        (row['count'] / species2_counts[row[f\"dist_category_{species_2}\"]]) * 100,\n",
    "        axis=1)\n",
    "    df['Harmonic_Mean'] = 2 / ((1 / df['Species1_Percentage']) + (1 / df['Species2_Percentage']))\n",
    "\n",
    "    heatmap_data = df.pivot(columns=f\"dist_category_{species_1}\",\n",
    "                            index=f\"dist_category_{species_2}\",\n",
    "                            values='Harmonic_Mean')\n",
    "\n",
    "    heatmap_data = np.round(heatmap_data.astype(\"float32\"), 2)\n",
    "\n",
    "\n",
    "  \n",
    "    order = ['expressed in over 90%',\n",
    "                 'expressed in over 30%',\n",
    "                 'expressed in less than 30%',\n",
    "                 'expressed in single',\n",
    "                 'lowly expressed']\n",
    "\n",
    "    fig = px.imshow(heatmap_data[order].reindex(order), title='One2one orthologs',\n",
    "                    text_auto=True,\n",
    "                    width=750, height=600,\n",
    "                    labels=dict(x=species_1, y=species_2, color=\"% overlap\"))\n",
    "    fig['layout']['yaxis']['autorange'] = \"reversed\"\n",
    "    fig.update_yaxes(autorange=True)\n",
    "\n",
    "    return heatmap_data, fig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "f6682e21-4af7-41f3-8877-f3ed0fc4513b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finished drerio, xtropicalis\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "mapping_both = pd.read_csv(f\"homology_mapped_o2o_{species_1}_{species_2}_1TPM.csv\")\n",
    "\n",
    "heatmap_data, o2o_heatmap = plot_cross_species_dist_category_heatmap(species_1=species_1, species_2=species_2, mapping_both=mapping_both)\n",
    "\n",
    "o2o_heatmap.write_image(f\"{species_1}_{species_2}_o2o_heatmap_dist_1TPM.pdf\")\n",
    "\n",
    "heatmap_data.to_csv(f\"o2o_heatmap_data_dist_{species_1}_{species_2}_1TPM.csv\")\n",
    "\n",
    "print(f\"finished {species_1}, {species_2}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9d43d62-1eaf-407b-b4fd-baf62a6e9bab",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "metacells_py",
   "language": "python",
   "name": "metacells_py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
